--- eventum-3.6.1-76-g74f125a7a/vendor/composer/autoload_classmap.php~	2019-03-07 17:19:22.676504958 +0200
+++ eventum-3.6.1-76-g74f125a7a/vendor/composer/autoload_classmap.php	2019-03-07 17:20:50.191892269 +0200
@@ -4,6 +4,7 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     'APIAuthToken' => $baseDir . '/lib/eventum/auth/APIAuthToken.php',
@@ -62,8 +62,6 @@
     'News' => $baseDir . '/lib/eventum/class.news.php',
     'Note' => $baseDir . '/lib/eventum/class.note.php',
     'Notification' => $baseDir . '/lib/eventum/class.notification.php',
-    'PHPlot' => $vendorDir . '/phplot/phplot/phplot.php',
-    'PHPlot_truecolor' => $vendorDir . '/phplot/phplot/phplot.php',
     'Pager' => $baseDir . '/lib/eventum/class.pager.php',
     'Partner' => $baseDir . '/lib/eventum/class.partner.php',
     'Phone_Support' => $baseDir . '/lib/eventum/class.phone_support.php',
@@ -94,12 +92,9 @@
     'SebastianBergmann\\Diff\\TimeEfficientLongestCommonSubsequenceCalculator' => $vendorDir . '/sebastian/diff/src/TimeEfficientLongestCommonSubsequenceCalculator.php',
     'Setup' => $baseDir . '/lib/eventum/class.setup.php',
     'Severity' => $baseDir . '/lib/eventum/class.severity.php',
-    'SphinxClient' => $vendorDir . '/sphinx/php-sphinxapi/sphinxapi.php',
     'Sphinx_Fulltext_Search' => $baseDir . '/lib/eventum/search/class.sphinx_fulltext_search.php',
     'Stats' => $baseDir . '/lib/eventum/class.stats.php',
     'Status' => $baseDir . '/lib/eventum/class.status.php',
-    'StreamReader' => $vendorDir . '/php-gettext/php-gettext/streams.php',
-    'StringReader' => $vendorDir . '/php-gettext/php-gettext/streams.php',
     'Support' => $baseDir . '/lib/eventum/class.support.php',
     'SupportLevelNotFoundException' => $baseDir . '/lib/eventum/crm/class.support_level.php',
     'Support_Level' => $baseDir . '/lib/eventum/crm/class.support_level.php',
@@ -110,3 +105,7 @@
     'Workflow' => $baseDir . '/lib/eventum/class.workflow.php',
-    'gettext_reader' => $vendorDir . '/php-gettext/php-gettext/gettext.php',
+
+    // external deps
+    'PHPlot' => $phpDir .'/phplot/phplot.php',
+    'Smarty' => $phpDir. '/Smarty3/Smarty.class.php',
+    'SphinxClient' => $phpDir . '/sphinxapi.php',
 );
--- eventum-3.6.5-35-g379513d37/vendor/composer/autoload_files.php~	2019-04-18 13:05:11.000000000 +0300
+++ eventum-3.6.5-35-g379513d37/vendor/composer/autoload_files.php	2019-04-18 14:27:33.184919302 +0300
@@ -4,14 +4,13 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     '72142d7b40a3a0b14e91825290b5ad82' => $vendorDir . '/cakephp/core/functions.php',
     '028fdea3165c4ba1ecccc83b7fec69fc' => $vendorDir . '/cakephp/collection/functions.php',
     '948ad5488880985ff1c06721a4e447fe' => $vendorDir . '/cakephp/utility/bootstrap.php',
     '2cffec82183ee1cea088009cef9a6fc3' => $vendorDir . '/ezyang/htmlpurifier/library/HTMLPurifier.composer.php',
-    'f084d01b0a599f67676cffef638aa95b' => $vendorDir . '/smarty/smarty/libs/bootstrap.php',
-    'b2b3f01361f3aaeab67fde08ae76446f' => $vendorDir . '/smarty-gettext/smarty-gettext/block.t.php',
-    'd3114450b53a83020e2822546c24e1ad' => $vendorDir . '/smarty-gettext/smarty-gettext/function.locale.php',
     'd12685060d57775f2bc3adf90c7d62bd' => $baseDir . '/lib/eventum/gettext.php',
+    'b2b3f01361f3aaeab67fde08ae76446f' => $phpDir. '/Smarty3/plugins/block.t.php',
 );
--- eventum-3.6.5/vendor/composer/autoload_psr4.php~	2019-04-08 13:06:08.567069724 +0300
+++ eventum-3.6.5/vendor/composer/autoload_psr4.php	2019-04-08 13:06:31.383421815 +0300
@@ -4,18 +4,11 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     'malkusch\\lock\\' => array($vendorDir . '/malkusch/lock/classes'),
-    'glen\\FilenameNormalizer\\Test\\' => array($vendorDir . '/glen/filename-normalizer/tests'),
     'glen\\FilenameNormalizer\\' => array($vendorDir . '/glen/filename-normalizer/src'),
-    'Zend\\Validator\\' => array($vendorDir . '/zendframework/zend-validator/src'),
-    'Zend\\Stdlib\\' => array($vendorDir . '/zendframework/zend-stdlib/src'),
-    'Zend\\ServiceManager\\' => array($vendorDir . '/zendframework/zend-servicemanager/src'),
-    'Zend\\Mime\\' => array($vendorDir . '/zendframework/zend-mime/src'),
-    'Zend\\Mail\\' => array($vendorDir . '/zendframework/zend-mail/src'),
-    'Zend\\Loader\\' => array($vendorDir . '/zendframework/zend-loader/src'),
-    'Zend\\Config\\' => array($vendorDir . '/zendframework/zend-config/src'),
     'Webuni\\CommonMark\\TableExtension\\' => array($vendorDir . '/webuni/commonmark-table-extension/src'),
     'TrueBV\\' => array($vendorDir . '/true/punycode/src'),
     'Symfony\\Contracts\\' => array($vendorDir . '/symfony/contracts'),
@@ -25,12 +25,9 @@
     'Symfony\\Component\\Console\\' => array($vendorDir . '/symfony/console'),
     'Symfony\\Component\\Config\\' => array($vendorDir . '/symfony/config'),
     'Silly\\' => array($vendorDir . '/mnapoli/silly/src'),
-    'Psr\\Log\\' => array($vendorDir . '/psr/log/Psr/Log'),
-    'Psr\\Container\\' => array($vendorDir . '/psr/container/src'),
     'PhpXmlRpc\\' => array($vendorDir . '/phpxmlrpc/phpxmlrpc/src'),
     'Phlib\\Flysystem\\Pdo\\' => array($vendorDir . '/phlib/flysystem-pdo/src'),
     'Phinx\\' => array($vendorDir . '/robmorgan/phinx/src/Phinx'),
-    'Monolog\\' => array($vendorDir . '/monolog/monolog/src/Monolog'),
     'Lossendae\\CommonMark\\TaskLists\\' => array($vendorDir . '/lossendae/commonmark-task-lists/src'),
     'League\\Flysystem\\' => array($vendorDir . '/league/flysystem/src'),
     'League\\CommonMark\\Ext\\InlinesOnly\\' => array($vendorDir . '/league/commonmark-ext-inlines-only/src'),
@@ -34,7 +34,6 @@
     'League\\CommonMark\\Ext\\Autolink\\' => array($vendorDir . '/league/commonmark-ext-autolink/src'),
     'League\\CommonMark\\' => array($vendorDir . '/league/commonmark/src'),
     'Invoker\\' => array($vendorDir . '/php-di/invoker/src'),
-    'Eventum\\Test\\' => array($baseDir . '/tests'),
     'Eventum\\' => array($baseDir . '/src'),
     'EmailReplyParser\\' => array($vendorDir . '/willdurand/email-reply-parser/src/EmailReplyParser'),
     'Ds\\' => array($vendorDir . '/php-ds/php-ds/src'),
@@ -60,2 +50,12 @@
     'Cake\\Cache\\' => array($vendorDir . '/cakephp/cache'),
+
+    'Zend\\Validator\\' =>  array($phpDir . '/Zend/Validator'),
+    'Zend\\ServiceManager\\' => array($phpDir . '/Zend/ServiceManager'),
+    'Zend\\Stdlib\\' => array($phpDir . '/Zend/Stdlib'),
+    'Zend\\Mime\\' => array($phpDir . '/Zend/Mime'),
+    'Zend\\Mail\\' => array($phpDir . '/Zend/Mail'),
+    'Zend\\Loader\\' => array($phpDir . '/Zend/Loader'),
+    'Zend\\Config\\' => array($phpDir . '/Zend/Config'),
+    'Monolog\\' => array($phpDir . '/Monolog'),
+    'Psr\\Log\\' => array($phpDir. '/Psr/Log'),
 );
--- eventum-3.0.3-13-gfefa325/autoload.php~	2015-10-13 21:36:40.000000000 +0300
+++ eventum-3.0.3-13-gfefa325/autoload.php	2015-10-13 23:39:26.450266254 +0300
@@ -48,4 +48,4 @@
 define('APP_PHP_GETTEXT_PATH', APP_PATH . '/vendor/php-gettext/php-gettext');
 
 // fonts directory for phplot
-define('APP_FONTS_PATH', APP_PATH . '/vendor/fonts/liberation');
+define('APP_FONTS_PATH', '/usr/share/fonts/TTF');
--- eventum-3.6.3-29-g154295bd2/vendor/composer/autoload_real.php~	2019-03-26 09:59:16.000000000 +0200
+++ eventum-3.6.3-29-g154295bd2/vendor/composer/autoload_real.php	2019-03-26 13:18:14.402093309 +0200
@@ -23,7 +23,7 @@
         self::$loader = $loader = new \Composer\Autoload\ClassLoader();
         spl_autoload_unregister(array('ComposerAutoloaderInitEventumCore', 'loadClassLoader'));
 
-        $useStaticLoader = PHP_VERSION_ID >= 50600 && !defined('HHVM_VERSION') && (!function_exists('zend_loader_file_encoded') || !zend_loader_file_encoded());
+        $useStaticLoader = false; // autoload_static.php patching is annoying and not worth the effort
         if ($useStaticLoader) {
             require_once __DIR__ . '/autoload_static.php';
 
