--- eventum-3.1.7/vendor/composer/autoload_classmap.php~	2016-12-28 18:45:59.299156669 +0200
+++ eventum-3.1.7/vendor/composer/autoload_classmap.php	2016-12-28 18:47:18.984639485 +0200
@@ -4,6 +4,7 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     'APIAuthToken' => $baseDir . '/lib/eventum/auth/APIAuthToken.php',
@@ -24,7 +25,6 @@
     'CLI_Misc' => $baseDir . '/lib/eventum/class.cli_misc.php',
     'CRM' => $baseDir . '/lib/eventum/class.crm.php',
     'CRMException' => $baseDir . '/lib/eventum/class.crmexception.php',
-    'CachedFileReader' => $vendorDir . '/php-gettext/php-gettext/streams.php',
     'Category' => $baseDir . '/lib/eventum/class.category.php',
     'Command_Line' => $baseDir . '/lib/eventum/class.command_line.php',
     'Contact' => $baseDir . '/lib/eventum/crm/class.contact.php',
@@ -45,8 +45,7 @@
     'Email_Response' => $baseDir . '/lib/eventum/class.email_response.php',
     'Eventum_Bot' => $baseDir . '/lib/eventum/irc/Eventum_Bot.php',
     'FAQ' => $baseDir . '/lib/eventum/class.faq.php',
-    'FileReader' => $vendorDir . '/php-gettext/php-gettext/streams.php',
     'Filter' => $baseDir . '/lib/eventum/class.filter.php',
     'Group' => $baseDir . '/lib/eventum/class.group.php',
     'Help' => $baseDir . '/lib/eventum/class.help.php',
     'History' => $baseDir . '/lib/eventum/class.history.php',
@@ -59,44 +58,14 @@
     'Lock' => $baseDir . '/lib/eventum/class.lock.php',
     'Mail_Helper' => $baseDir . '/lib/eventum/class.mail_helper.php',
     'Mail_Queue' => $baseDir . '/lib/eventum/class.mail_queue.php',
-    'Mail_mime' => $vendorDir . '/pear-pear.php.net/Mail_Mime/Mail/mime.php',
-    'Mail_mimeDecode' => $vendorDir . '/pear-pear.php.net/Mail_mimeDecode/Mail/mimeDecode.php',
-    'Mail_mimePart' => $vendorDir . '/pear-pear.php.net/Mail_Mime/Mail/mimePart.php',
-    'Math_Stats' => $vendorDir . '/pear-pear.php.net/Math_Stats/Math/Stats.php',
     'Mime_Helper' => $baseDir . '/lib/eventum/class.mime_helper.php',
     'Misc' => $baseDir . '/lib/eventum/class.misc.php',
     'Monitor' => $baseDir . '/lib/eventum/class.monitor.php',
     'MySQL_Fulltext_Search' => $baseDir . '/lib/eventum/search/class.mysql_fulltext_search.php',
     'Mysql_Auth_Backend' => $baseDir . '/lib/eventum/auth/class.mysql_auth_backend.php',
-    'Net_LDAP2' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2.php',
-    'Net_LDAP2_Entry' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2/Entry.php',
-    'Net_LDAP2_Error' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2.php',
-    'Net_LDAP2_Filter' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2/Filter.php',
-    'Net_LDAP2_LDIF' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2/LDIF.php',
-    'Net_LDAP2_RootDSE' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2/RootDSE.php',
-    'Net_LDAP2_Schema' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2/Schema.php',
-    'Net_LDAP2_SchemaCache' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2/SchemaCache.interface.php',
-    'Net_LDAP2_Search' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2/Search.php',
-    'Net_LDAP2_SimpleFileSchemaCache' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2/SimpleFileSchemaCache.php',
-    'Net_LDAP2_Util' => $vendorDir . '/pear-pear.php.net/Net_LDAP2/Net/LDAP2/Util.php',
-    'Net_POP3' => $vendorDir . '/pear-pear.php.net/Net_POP3/Net/POP3.php',
-    'Net_SmartIRC' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC.php',
-    'Net_SmartIRC_Error' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC.php',
-    'Net_SmartIRC_channel' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC.php',
-    'Net_SmartIRC_channeluser' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC.php',
-    'Net_SmartIRC_data' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC.php',
-    'Net_SmartIRC_irccommands' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC/irccommands.php',
-    'Net_SmartIRC_ircuser' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC.php',
-    'Net_SmartIRC_listenfor' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC.php',
-    'Net_SmartIRC_messagehandler' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC/messagehandler.php',
-    'Net_SmartIRC_user' => $vendorDir . '/pear-pear.php.net/Net_SmartIRC/Net/SmartIRC.php',
-    'Net_Socket' => $vendorDir . '/pear-pear.php.net/Net_Socket/Net/Socket.php',
-    'Net_URL' => $vendorDir . '/pear-pear.php.net/Net_URL/Net/URL.php',
     'News' => $baseDir . '/lib/eventum/class.news.php',
     'Note' => $baseDir . '/lib/eventum/class.note.php',
     'Notification' => $baseDir . '/lib/eventum/class.notification.php',
-    'PHPlot' => $vendorDir . '/phplot/phplot/phplot.php',
-    'PHPlot_truecolor' => $vendorDir . '/phplot/phplot/phplot.php',
     'Pager' => $baseDir . '/lib/eventum/class.pager.php',
     'Partner' => $baseDir . '/lib/eventum/class.partner.php',
     'Phone_Support' => $baseDir . '/lib/eventum/class.phone_support.php',
@@ -109,6 +78,8 @@
     'Reminder' => $baseDir . '/lib/eventum/class.reminder.php',
     'Reminder_Action' => $baseDir . '/lib/eventum/class.reminder_action.php',
     'Reminder_Condition' => $baseDir . '/lib/eventum/class.reminder_condition.php',
+    'RemoteApi' => $baseDir . '/lib/eventum/rpc/RemoteApi.php',
+    'RemoteApiException' => $baseDir . '/lib/eventum/rpc/RemoteApi.php',
     'Report' => $baseDir . '/lib/eventum/class.report.php',
     'Reporter_List_Custom_Field_Backend' => $baseDir . '/lib/eventum/custom_field/class.reporter_list.php',
     'Resolution' => $baseDir . '/lib/eventum/class.resolution.php',
@@ -120,43 +91,21 @@
     'SessionHandlerInterface' => $vendorDir . '/symfony/http-foundation/Resources/stubs/SessionHandlerInterface.php',
     'Setup' => $baseDir . '/lib/eventum/class.setup.php',
     'Severity' => $baseDir . '/lib/eventum/class.severity.php',
-    'SphinxClient' => $vendorDir . '/sphinx/php-sphinxapi/sphinxapi.php',
     'Sphinx_Fulltext_Search' => $baseDir . '/lib/eventum/search/class.sphinx_fulltext_search.php',
     'Stats' => $baseDir . '/lib/eventum/class.stats.php',
     'Status' => $baseDir . '/lib/eventum/class.status.php',
-    'StreamReader' => $vendorDir . '/php-gettext/php-gettext/streams.php',
-    'StringReader' => $vendorDir . '/php-gettext/php-gettext/streams.php',
     'Support' => $baseDir . '/lib/eventum/class.support.php',
     'SupportLevelNotFoundException' => $baseDir . '/lib/eventum/crm/class.support_level.php',
     'Support_Level' => $baseDir . '/lib/eventum/crm/class.support_level.php',
     'Template_Helper' => $baseDir . '/lib/eventum/class.template_helper.php',
-    'Text_Diff' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff3' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff3.php',
-    'Text_Diff3_BlockBuilder' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff3.php',
-    'Text_Diff3_Op' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff3.php',
-    'Text_Diff3_Op_copy' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff3.php',
-    'Text_Diff_Engine_native' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Engine/native.php',
-    'Text_Diff_Engine_shell' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Engine/shell.php',
-    'Text_Diff_Engine_string' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Engine/string.php',
-    'Text_Diff_Engine_xdiff' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Engine/xdiff.php',
-    'Text_Diff_Mapped' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Mapped.php',
-    'Text_Diff_Op' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Op_add' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Op_change' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Op_copy' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Op_delete' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Renderer' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Renderer.php',
-    'Text_Diff_Renderer_context' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Renderer/context.php',
-    'Text_Diff_Renderer_inline' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Renderer/inline.php',
-    'Text_Diff_Renderer_unified' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Renderer/unified.php',
-    'Text_Diff_ThreeWay' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/ThreeWay.php',
-    'Text_Diff_ThreeWay_BlockBuilder' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/ThreeWay.php',
-    'Text_Diff_ThreeWay_Op' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/ThreeWay.php',
-    'Text_Diff_ThreeWay_Op_copy' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/ThreeWay.php',
-    'Text_MappedDiff' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
     'Time_Tracking' => $baseDir . '/lib/eventum/class.time_tracking.php',
     'User' => $baseDir . '/lib/eventum/class.user.php',
     'Validation' => $baseDir . '/lib/eventum/class.validation.php',
     'Workflow' => $baseDir . '/lib/eventum/class.workflow.php',
-    'gettext_reader' => $vendorDir . '/php-gettext/php-gettext/gettext.php',
+    'XmlRpcServer' => $baseDir . '/lib/eventum/rpc/XmlRpcServer.php',
+
+    // external deps
+    'PHPlot' => $phpDir .'/phplot/phplot.php',
+    'Smarty' => $phpDir. '/Smarty3/Smarty.class.php',
+    'SphinxClient' => $phpDir . '/sphinxapi.php',
 );
--- eventum-3.1.7/vendor/composer/autoload_files.php~	2016-12-28 18:37:10.000000000 +0200
+++ eventum-3.1.7/vendor/composer/autoload_files.php	2016-12-28 18:52:04.467065215 +0200
@@ -4,11 +4,10 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
-    'f084d01b0a599f67676cffef638aa95b' => $vendorDir . '/smarty/smarty/libs/bootstrap.php',
     '8170285c807a9f24f165f37b15bc9a36' => $vendorDir . '/defuse/php-encryption/Crypto.php',
-    'b2b3f01361f3aaeab67fde08ae76446f' => $vendorDir . '/smarty-gettext/smarty-gettext/block.t.php',
-    'd3114450b53a83020e2822546c24e1ad' => $vendorDir . '/smarty-gettext/smarty-gettext/function.locale.php',
     'd12685060d57775f2bc3adf90c7d62bd' => $baseDir . '/lib/eventum/gettext.php',
+    'b2b3f01361f3aaeab67fde08ae76446f' => $phpDir. '/Smarty3/plugins/block.t.php',
 );
--- eventum-3.1.3/vendor/composer/autoload_namespaces.php~	2016-09-25 21:27:02.000000000 +0300
+++ eventum-3.1.3/vendor/composer/autoload_namespaces.php	2016-09-25 21:51:42.028861261 +0300
@@ -3,15 +3,17 @@
 // autoload_namespaces.php @generated by Composer
 
 $vendorDir = dirname(dirname(__FILE__));
-$baseDir = dirname($vendorDir);
+$pearDir = PEAR_INSTALL_DIR . '/';
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     'SecurityLib' => array($vendorDir . '/ircmaxell/security-lib/lib'),
     'RandomLib' => array($vendorDir . '/ircmaxell/random-lib/lib'),
-    'PEAR' => array($vendorDir . '/pear/pear_exception'),
-    'Net' => array($vendorDir . '/pear/net_smtp'),
-    'Mail' => array($vendorDir . '/pear/mail'),
-    'DB' => array($vendorDir . '/pear/db'),
-    'Auth' => array($vendorDir . '/pear/auth_sasl'),
-    '' => array($vendorDir . '/pear/pear-core-minimal/src'),
-);
+    'Auth' => array($pearDir),
+    'DB' => array($pearDir),
+    'Mail' => array($pearDir),
+    'Net' => array($pearDir),
+    'PEAR' => array($pearDir),
+    'Text' => array($pearDir),
+    'XML' => array($pearDir),
+);
--- eventum-3.1.7/vendor/composer/autoload_psr4.php~	2016-12-28 18:37:10.000000000 +0200
+++ eventum-3.1.7/vendor/composer/autoload_psr4.php	2016-12-28 18:53:20.699266184 +0200
@@ -6,22 +6,22 @@
 $baseDir = dirname($vendorDir);
 
 return array(
+    'Zend\\Validator\\' =>  array($phpDir . '/Zend/Validator'),
+    'Zend\\Stdlib\\' => array($phpDir . '/Zend/Stdlib'),
+    'Zend\\Mime\\' => array($phpDir . '/Zend/Mime'),
+    'Zend\\Mail\\' => array($phpDir . '/Zend/Mail'),
+    'Zend\\Loader\\' => array($phpDir . '/Zend/Loader'),
+    'Zend\\Config\\' => array($phpDir . '/Zend/Config'),
+    'Symfony\\Component\\Yaml\\' => array($phpDir . '/Symfony/Component/Yaml'),
+    'Symfony\\Component\\Serializer\\' => array($phpDir . '/Symfony/Component/Serializer'),
+    'Symfony\\Component\\OptionsResolver\\' =>  array($phpDir . '/Symfony/Component/OptionsResolver'),
+    'Symfony\\Component\\HttpFoundation\\' => array($phpDir . '/Symfony/Component/HttpFoundation'),
+    'Symfony\\Component\\Filesystem\\' => array($phpDir . '/Symfony/Component/Filesystem'),
+    'Symfony\\Component\\Config\\' =>  array($phpDir . '/Symfony/Component/Config'),
+    'Monolog\\' => array($phpDir . '/Monolog'),
+    'Psr\\Log\\' => array($phpDir. '/Psr/Log'),
     'malkusch\\lock\\' => array($vendorDir . '/malkusch/lock/classes'),
-    'Zend\\Validator\\' => array($vendorDir . '/zendframework/zend-validator/src'),
-    'Zend\\Stdlib\\' => array($vendorDir . '/zendframework/zend-stdlib/src'),
-    'Zend\\Mime\\' => array($vendorDir . '/zendframework/zend-mime/src'),
-    'Zend\\Mail\\' => array($vendorDir . '/zendframework/zend-mail/src'),
-    'Zend\\Loader\\' => array($vendorDir . '/zendframework/zend-loader/src'),
-    'Zend\\Config\\' => array($vendorDir . '/zendframework/zend-config/src'),
-    'Symfony\\Component\\Yaml\\' => array($vendorDir . '/symfony/yaml'),
-    'Symfony\\Component\\Serializer\\' => array($vendorDir . '/symfony/serializer'),
-    'Symfony\\Component\\OptionsResolver\\' => array($vendorDir . '/symfony/options-resolver'),
-    'Symfony\\Component\\HttpFoundation\\' => array($vendorDir . '/symfony/http-foundation'),
-    'Symfony\\Component\\Filesystem\\' => array($vendorDir . '/symfony/filesystem'),
-    'Symfony\\Component\\Config\\' => array($vendorDir . '/symfony/config'),
-    'Psr\\Log\\' => array($vendorDir . '/psr/log/Psr/Log'),
     'PhpXmlRpc\\' => array($vendorDir . '/phpxmlrpc/phpxmlrpc/src'),
-    'Monolog\\' => array($vendorDir . '/monolog/monolog/src/Monolog'),
     'Eventum\\' => array($baseDir . '/src'),
     'EmailReplyParser\\' => array($vendorDir . '/willdurand/email-reply-parser/src/EmailReplyParser'),
     'Cascade\\' => array($vendorDir . '/theorchard/monolog-cascade/src'),
--- eventum-3.0.3-13-gfefa325/autoload.php~	2015-10-13 21:36:40.000000000 +0300
+++ eventum-3.0.3-13-gfefa325/autoload.php	2015-10-13 23:39:26.450266254 +0300
@@ -48,4 +48,4 @@
 define('APP_PHP_GETTEXT_PATH', APP_PATH . '/vendor/php-gettext/php-gettext');
 
 // fonts directory for phplot
-define('APP_FONTS_PATH', APP_PATH . '/vendor/fonts/liberation');
+define('APP_FONTS_PATH', '/usr/share/fonts/TTF');
--- eventum-3.1.6/vendor/composer/autoload_real.php~	2016-12-09 01:12:28.646671725 +0200
+++ eventum-3.1.6/vendor/composer/autoload_real.php	2016-12-09 01:12:30.253314754 +0200
@@ -23,11 +23,7 @@
         self::$loader = $loader = new \Composer\Autoload\ClassLoader();
         spl_autoload_unregister(array('ComposerAutoloaderInitEventumCore', 'loadClassLoader'));
 
-        $includePaths = require __DIR__ . '/include_paths.php';
-        array_push($includePaths, get_include_path());
-        set_include_path(implode(PATH_SEPARATOR, $includePaths));
-
-        $useStaticLoader = PHP_VERSION_ID >= 50600 && !defined('HHVM_VERSION') && (!function_exists('zend_loader_file_encoded') || !zend_loader_file_encoded());
+        $useStaticLoader = false; // TODO: patch autoload_static.php as well
         if ($useStaticLoader) {
             require_once __DIR__ . '/autoload_static.php';
 
