--- eventum-3.5.6-75-g293a8feb/vendor/composer/autoload_classmap.php~	2019-01-23 00:18:07.562769763 +0200
+++ eventum-3.5.6-75-g293a8feb/vendor/composer/autoload_classmap.php	2019-01-23 00:19:46.887983481 +0200
@@ -4,6 +4,7 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     'APIAuthToken' => $baseDir . '/lib/eventum/auth/APIAuthToken.php',
@@ -48,19 +48,13 @@
     'Link_Filter' => $baseDir . '/lib/eventum/class.link_filter.php',
     'Mail_Helper' => $baseDir . '/lib/eventum/class.mail_helper.php',
     'Mail_Queue' => $baseDir . '/lib/eventum/class.mail_queue.php',
-    'Math_Stats' => $vendorDir . '/pear-pear.php.net/Math_Stats/Math/Stats.php',
     'Mime_Helper' => $baseDir . '/lib/eventum/class.mime_helper.php',
     'Misc' => $baseDir . '/lib/eventum/class.misc.php',
     'MySQL_Fulltext_Search' => $baseDir . '/lib/eventum/search/class.mysql_fulltext_search.php',
-    'Net_POP3' => $vendorDir . '/pear-pear.php.net/Net_POP3/Net/POP3.php',
-    'Net_Socket' => $vendorDir . '/pear-pear.php.net/Net_Socket/Net/Socket.php',
-    'Net_URL' => $vendorDir . '/pear-pear.php.net/Net_URL/Net/URL.php',
     'News' => $baseDir . '/lib/eventum/class.news.php',
     'Normalizer' => $vendorDir . '/symfony/polyfill-intl-normalizer/Resources/stubs/Normalizer.php',
     'Note' => $baseDir . '/lib/eventum/class.note.php',
     'Notification' => $baseDir . '/lib/eventum/class.notification.php',
-    'PHPlot' => $vendorDir . '/phplot/phplot/phplot.php',
-    'PHPlot_truecolor' => $vendorDir . '/phplot/phplot/phplot.php',
     'Pager' => $baseDir . '/lib/eventum/class.pager.php',
     'Partner' => $baseDir . '/lib/eventum/class.partner.php',
     'Phone_Support' => $baseDir . '/lib/eventum/class.phone_support.php',
@@ -83,43 +77,20 @@
     'Search_Profile' => $baseDir . '/lib/eventum/class.search_profile.php',
     'Setup' => $baseDir . '/lib/eventum/class.setup.php',
     'Severity' => $baseDir . '/lib/eventum/class.severity.php',
-    'SphinxClient' => $vendorDir . '/sphinx/php-sphinxapi/sphinxapi.php',
     'Sphinx_Fulltext_Search' => $baseDir . '/lib/eventum/search/class.sphinx_fulltext_search.php',
     'Stats' => $baseDir . '/lib/eventum/class.stats.php',
     'Status' => $baseDir . '/lib/eventum/class.status.php',
-    'StreamReader' => $vendorDir . '/php-gettext/php-gettext/streams.php',
-    'StringReader' => $vendorDir . '/php-gettext/php-gettext/streams.php',
     'Support' => $baseDir . '/lib/eventum/class.support.php',
     'SupportLevelNotFoundException' => $baseDir . '/lib/eventum/crm/class.support_level.php',
     'Support_Level' => $baseDir . '/lib/eventum/crm/class.support_level.php',
     'Template_Helper' => $baseDir . '/lib/eventum/class.template_helper.php',
-    'Text_Diff' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff3' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff3.php',
-    'Text_Diff3_BlockBuilder' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff3.php',
-    'Text_Diff3_Op' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff3.php',
-    'Text_Diff3_Op_copy' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff3.php',
-    'Text_Diff_Engine_native' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Engine/native.php',
-    'Text_Diff_Engine_shell' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Engine/shell.php',
-    'Text_Diff_Engine_string' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Engine/string.php',
-    'Text_Diff_Engine_xdiff' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Engine/xdiff.php',
-    'Text_Diff_Mapped' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Mapped.php',
-    'Text_Diff_Op' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Op_add' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Op_change' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Op_copy' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Op_delete' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
-    'Text_Diff_Renderer' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Renderer.php',
-    'Text_Diff_Renderer_context' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Renderer/context.php',
-    'Text_Diff_Renderer_inline' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Renderer/inline.php',
-    'Text_Diff_Renderer_unified' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/Renderer/unified.php',
-    'Text_Diff_ThreeWay' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/ThreeWay.php',
-    'Text_Diff_ThreeWay_BlockBuilder' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/ThreeWay.php',
-    'Text_Diff_ThreeWay_Op' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/ThreeWay.php',
-    'Text_Diff_ThreeWay_Op_copy' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff/ThreeWay.php',
-    'Text_MappedDiff' => $vendorDir . '/pear-pear.php.net/Text_Diff/Text/Diff.php',
     'Time_Tracking' => $baseDir . '/lib/eventum/class.time_tracking.php',
     'User' => $baseDir . '/lib/eventum/class.user.php',
     'Validation' => $baseDir . '/lib/eventum/class.validation.php',
     'Workflow' => $baseDir . '/lib/eventum/class.workflow.php',
-    'gettext_reader' => $vendorDir . '/php-gettext/php-gettext/gettext.php',
+
+    // external deps
+    'PHPlot' => $phpDir .'/phplot/phplot.php',
+    'Smarty' => $phpDir. '/Smarty3/Smarty.class.php',
+    'SphinxClient' => $phpDir . '/sphinxapi.php',
 );
--- eventum-3.5.6-75-g293a8feb/vendor/composer/autoload_files.php~	2019-01-22 23:56:16.000000000 +0200
+++ eventum-3.5.6-75-g293a8feb/vendor/composer/autoload_files.php	2019-01-23 00:22:07.852588625 +0200
@@ -4,15 +4,13 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
-    '320cde22f66dd4f5d3fd621d3e88b98f' => $vendorDir . '/symfony/polyfill-ctype/bootstrap.php',
     '72142d7b40a3a0b14e91825290b5ad82' => $vendorDir . '/cakephp/core/functions.php',
     '028fdea3165c4ba1ecccc83b7fec69fc' => $vendorDir . '/cakephp/collection/functions.php',
     '948ad5488880985ff1c06721a4e447fe' => $vendorDir . '/cakephp/utility/bootstrap.php',
-    'f084d01b0a599f67676cffef638aa95b' => $vendorDir . '/smarty/smarty/libs/bootstrap.php',
     'e69f7f6ee287b969198c3c9d6777bd38' => $vendorDir . '/symfony/polyfill-intl-normalizer/bootstrap.php',
-    'b2b3f01361f3aaeab67fde08ae76446f' => $vendorDir . '/smarty-gettext/smarty-gettext/block.t.php',
-    'd3114450b53a83020e2822546c24e1ad' => $vendorDir . '/smarty-gettext/smarty-gettext/function.locale.php',
     'd12685060d57775f2bc3adf90c7d62bd' => $baseDir . '/lib/eventum/gettext.php',
+    'b2b3f01361f3aaeab67fde08ae76446f' => $phpDir. '/Smarty3/plugins/block.t.php',
 );
--- eventum-3.5.6-75-g293a8feb/vendor/composer/autoload_namespaces.php~	2019-01-22 23:56:16.000000000 +0200
+++ eventum-3.5.6-75-g293a8feb/vendor/composer/autoload_namespaces.php	2019-01-23 00:25:11.149906012 +0200
@@ -3,16 +3,20 @@
 // autoload_namespaces.php @generated by Composer
 
 $vendorDir = dirname(dirname(__FILE__));
-$baseDir = dirname($vendorDir);
+$pearDir = PEAR_INSTALL_DIR . '/';
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     'SecurityLib' => array($vendorDir . '/ircmaxell/security-lib/lib'),
     'RandomLib' => array($vendorDir . '/ircmaxell/random-lib/lib'),
-    'PEAR' => array($vendorDir . '/pear/pear_exception'),
-    'Net' => array($vendorDir . '/pear/net_smtp'),
     'Enrise\\' => array($vendorDir . '/enrise/urihelper/src'),
     'Doctrine\\Common\\Lexer\\' => array($vendorDir . '/doctrine/lexer/lib'),
     'Doctrine\\Common\\Collections\\' => array($vendorDir . '/doctrine/collections/lib'),
-    'Auth' => array($vendorDir . '/pear/auth_sasl'),
-    '' => array($vendorDir . '/pear/pear-core-minimal/src'),
+    // PEAR deps
+    'Auth' => array($pearDir),
+    'Mail' => array($pearDir),
+    'Math' => array($pearDir),
+    'Net' => array($pearDir),
+    'PEAR' => array($pearDir),
+    'Text' => array($pearDir),
 );
--- eventum-3.5.6-75-g293a8feb/vendor/composer/autoload_psr4.php~	2019-01-22 23:56:16.000000000 +0200
+++ eventum-3.5.6-75-g293a8feb/vendor/composer/autoload_psr4.php	2019-01-23 00:30:36.488478338 +0200
@@ -10,13 +10,6 @@
     'glen\\FilenameNormalizer\\Test\\' => array($vendorDir . '/glen/filename-normalizer/tests'),
     'glen\\FilenameNormalizer\\' => array($vendorDir . '/glen/filename-normalizer/src'),
     'cebe\\markdown\\' => array($vendorDir . '/cebe/markdown'),
-    'Zend\\Validator\\' => array($vendorDir . '/zendframework/zend-validator/src'),
-    'Zend\\Stdlib\\' => array($vendorDir . '/zendframework/zend-stdlib/src'),
-    'Zend\\ServiceManager\\' => array($vendorDir . '/zendframework/zend-servicemanager/src'),
-    'Zend\\Mime\\' => array($vendorDir . '/zendframework/zend-mime/src'),
-    'Zend\\Mail\\' => array($vendorDir . '/zendframework/zend-mail/src'),
-    'Zend\\Loader\\' => array($vendorDir . '/zendframework/zend-loader/src'),
-    'Zend\\Config\\' => array($vendorDir . '/zendframework/zend-config/src'),
     'Symfony\\Polyfill\\Intl\\Normalizer\\' => array($vendorDir . '/symfony/polyfill-intl-normalizer'),
     'Symfony\\Polyfill\\Ctype\\' => array($vendorDir . '/symfony/polyfill-ctype'),
     'Symfony\\Contracts\\' => array($vendorDir . '/symfony/contracts'),
@@ -32,15 +25,11 @@
     'Symfony\\Component\\Console\\' => array($vendorDir . '/symfony/console'),
     'Symfony\\Component\\Config\\' => array($vendorDir . '/symfony/config'),
     'Silly\\' => array($vendorDir . '/mnapoli/silly/src'),
-    'Psr\\Log\\' => array($vendorDir . '/psr/log/Psr/Log'),
-    'Psr\\Container\\' => array($vendorDir . '/psr/container/src'),
     'PhpXmlRpc\\' => array($vendorDir . '/phpxmlrpc/phpxmlrpc/src'),
     'Phlib\\Flysystem\\Pdo\\' => array($vendorDir . '/phlib/flysystem-pdo/src'),
     'Phinx\\' => array($vendorDir . '/robmorgan/phinx/src/Phinx'),
-    'Monolog\\' => array($vendorDir . '/monolog/monolog/src/Monolog'),
     'League\\Flysystem\\' => array($vendorDir . '/league/flysystem/src'),
     'Invoker\\' => array($vendorDir . '/php-di/invoker/src'),
-    'Eventum\\Test\\' => array($baseDir . '/tests', $baseDir . '/tests/src'),
     'Eventum\\' => array($baseDir . '/src'),
     'EmailReplyParser\\' => array($vendorDir . '/willdurand/email-reply-parser/src/EmailReplyParser'),
     'Doctrine\\ORM\\' => array($vendorDir . '/doctrine/orm/lib/Doctrine/ORM'),
@@ -59,4 +48,14 @@
     'Cake\\Core\\' => array($vendorDir . '/cakephp/core'),
     'Cake\\Collection\\' => array($vendorDir . '/cakephp/collection'),
     'Cake\\Cache\\' => array($vendorDir . '/cakephp/cache'),
+
+    'Zend\\Validator\\' =>  array($phpDir . '/Zend/Validator'),
+    'Zend\\ServiceManager\\' => array($phpDir . '/Zend/ServiceManager'),
+    'Zend\\Stdlib\\' => array($phpDir . '/Zend/Stdlib'),
+    'Zend\\Mime\\' => array($phpDir . '/Zend/Mime'),
+    'Zend\\Mail\\' => array($phpDir . '/Zend/Mail'),
+    'Zend\\Loader\\' => array($phpDir . '/Zend/Loader'),
+    'Zend\\Config\\' => array($phpDir . '/Zend/Config'),
+    'Monolog\\' => array($phpDir . '/Monolog'),
+    'Psr\\Log\\' => array($phpDir. '/Psr/Log'),
 );
--- eventum-3.0.3-13-gfefa325/autoload.php~	2015-10-13 21:36:40.000000000 +0300
+++ eventum-3.0.3-13-gfefa325/autoload.php	2015-10-13 23:39:26.450266254 +0300
@@ -48,4 +48,4 @@
 define('APP_PHP_GETTEXT_PATH', APP_PATH . '/vendor/php-gettext/php-gettext');
 
 // fonts directory for phplot
-define('APP_FONTS_PATH', APP_PATH . '/vendor/fonts/liberation');
+define('APP_FONTS_PATH', '/usr/share/fonts/TTF');
--- eventum-3.3.4-365-g65c5538f/vendor/composer/autoload_real.php~	2018-03-13 23:16:48.000000000 +0200
+++ eventum-3.3.4-365-g65c5538f/vendor/composer/autoload_real.php	2018-03-13 23:35:23.062702617 +0200
@@ -23,11 +23,7 @@
         self::$loader = $loader = new \Composer\Autoload\ClassLoader();
         spl_autoload_unregister(array('ComposerAutoloaderInitEventumCore', 'loadClassLoader'));
 
-        $includePaths = require __DIR__ . '/include_paths.php';
-        $includePaths[] = get_include_path();
-        set_include_path(implode(PATH_SEPARATOR, $includePaths));
-
-        $useStaticLoader = PHP_VERSION_ID >= 50600 && !defined('HHVM_VERSION') && (!function_exists('zend_loader_file_encoded') || !zend_loader_file_encoded());
+        $useStaticLoader = false; // autoload_static.php patching is annoying and not worth the effort
         if ($useStaticLoader) {
             require_once __DIR__ . '/autoload_static.php';
 
