--- eventum-3.8.6/vendor/composer/autoload_classmap.php~	2019-11-24 18:16:45.598780557 +0200
+++ eventum-3.8.6/vendor/composer/autoload_classmap.php	2019-11-24 18:18:22.087329736 +0200
@@ -4,6 +4,7 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     'APIAuthToken' => $baseDir . '/lib/eventum/auth/APIAuthToken.php',
@@ -62,8 +62,6 @@
     'News' => $baseDir . '/lib/eventum/class.news.php',
     'Note' => $baseDir . '/lib/eventum/class.note.php',
     'Notification' => $baseDir . '/lib/eventum/class.notification.php',
-    'PHPlot' => $vendorDir . '/phplot/phplot/phplot.php',
-    'PHPlot_truecolor' => $vendorDir . '/phplot/phplot/phplot.php',
     'Pager' => $baseDir . '/lib/eventum/class.pager.php',
     'Partner' => $baseDir . '/lib/eventum/class.partner.php',
     'Phone_Support' => $baseDir . '/lib/eventum/class.phone_support.php',
@@ -299,7 +299,6 @@
     'Smarty_Template_Source' => $vendorDir . '/smarty/smarty/libs/sysplugins/smarty_template_source.php',
     'Smarty_Undefined_Variable' => $vendorDir . '/smarty/smarty/libs/sysplugins/smarty_undefined_variable.php',
     'Smarty_Variable' => $vendorDir . '/smarty/smarty/libs/sysplugins/smarty_variable.php',
-    'SphinxClient' => $vendorDir . '/sphinx/php-sphinxapi/sphinxapi.php',
     'Sphinx_Fulltext_Search' => $baseDir . '/lib/eventum/search/class.sphinx_fulltext_search.php',
     'SqlFormatter' => $vendorDir . '/jdorn/sql-formatter/lib/SqlFormatter.php',
     'Stats' => $baseDir . '/lib/eventum/class.stats.php',
@@ -142,4 +141,9 @@
     'UserSeeder' => $baseDir . '/db/seeds/UserSeeder.php',
     'Validation' => $baseDir . '/lib/eventum/class.validation.php',
     'Workflow' => $baseDir . '/lib/eventum/class.workflow.php',
+
+    // external deps
+    'PHPlot' => $phpDir .'/phplot/phplot.php',
+    'Smarty' => $phpDir. '/Smarty3/Smarty.class.php',
+    'SphinxClient' => $phpDir . '/sphinxapi.php',
 );
--- eventum-3.8.6/vendor/composer/autoload_files.php~	2019-11-24 18:16:45.598780557 +0200
+++ eventum-3.8.6/vendor/composer/autoload_files.php	2019-11-24 18:19:26.599693077 +0200
@@ -4,6 +4,7 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     '25072dd6e2470089de65ae7bf11d3109' => $vendorDir . '/symfony/polyfill-php72/bootstrap.php',
@@ -12,7 +12,6 @@
     '028fdea3165c4ba1ecccc83b7fec69fc' => $vendorDir . '/cakephp/collection/functions.php',
     '948ad5488880985ff1c06721a4e447fe' => $vendorDir . '/cakephp/utility/bootstrap.php',
     '2cffec82183ee1cea088009cef9a6fc3' => $vendorDir . '/ezyang/htmlpurifier/library/HTMLPurifier.composer.php',
-    'b2b3f01361f3aaeab67fde08ae76446f' => $vendorDir . '/smarty-gettext/smarty-gettext/block.t.php',
-    'd3114450b53a83020e2822546c24e1ad' => $vendorDir . '/smarty-gettext/smarty-gettext/function.locale.php',
     'd12685060d57775f2bc3adf90c7d62bd' => $baseDir . '/lib/eventum/gettext.php',
+    'b2b3f01361f3aaeab67fde08ae76446f' => $phpDir. '/Smarty3/plugins/block.t.php',
 );
--- eventum-3.8.5/vendor/composer/autoload_psr4.php~	2019-10-26 04:19:31.788361290 +0300
+++ eventum-3.8.5/vendor/composer/autoload_psr4.php	2019-10-26 04:53:23.625455791 +0300
@@ -4,10 +4,10 @@
 
 $vendorDir = dirname(dirname(__FILE__));
 $baseDir = dirname($vendorDir);
+$phpDir = PHP_DATADIR . '/php';
 
 return array(
     'malkusch\\lock\\' => array($vendorDir . '/malkusch/lock/classes'),
-    'glen\\FilenameNormalizer\\Test\\' => array($vendorDir . '/glen/filename-normalizer/tests'),
     'glen\\FilenameNormalizer\\' => array($vendorDir . '/glen/filename-normalizer/src'),
     'Zend\\Validator\\' => array($vendorDir . '/zendframework/zend-validator/src'),
     'Zend\\Stdlib\\' => array($vendorDir . '/zendframework/zend-stdlib/src'),
@@ -51,7 +51,6 @@
     'Silly\\' => array($vendorDir . '/mnapoli/silly/src'),
     'Seld\\Signal\\' => array($vendorDir . '/seld/signal-handler/src'),
     'Psr\\SimpleCache\\' => array($vendorDir . '/psr/simple-cache/src'),
-    'Psr\\Log\\' => array($vendorDir . '/psr/log/Psr/Log'),
     'Psr\\Container\\' => array($vendorDir . '/psr/container/src'),
     'Psr\\Cache\\' => array($vendorDir . '/psr/cache/src'),
     'Port\\Tests\\' => array($vendorDir . '/portphp/portphp/tests'),
@@ -64,7 +64,6 @@
     'Phlib\\Flysystem\\Pdo\\' => array($vendorDir . '/phlib/flysystem-pdo/src'),
     'Phinx\\' => array($vendorDir . '/robmorgan/phinx/src/Phinx'),
     'PackageVersions\\' => array($vendorDir . '/ocramius/package-versions/src/PackageVersions'),
-    'Monolog\\' => array($vendorDir . '/monolog/monolog/src/Monolog'),
     'League\\HTMLToMarkdown\\' => array($vendorDir . '/league/html-to-markdown/src'),
     'League\\Flysystem\\' => array($vendorDir . '/league/flysystem/src'),
     'League\\CommonMark\\Ext\\TaskList\\' => array($vendorDir . '/league/commonmark-ext-task-list/src'),
@@ -45,7 +53,6 @@
     'League\\CommonMark\\' => array($vendorDir . '/league/commonmark/src'),
     'Invoker\\' => array($vendorDir . '/php-di/invoker/src'),
     'Interop\\Container\\' => array($vendorDir . '/container-interop/container-interop/src/Interop/Container'),
-    'Eventum\\Test\\' => array($baseDir . '/tests'),
     'Eventum\\' => array($baseDir . '/src'),
     'EmailReplyParser\\' => array($vendorDir . '/willdurand/email-reply-parser/src/EmailReplyParser'),
     'Ds\\' => array($vendorDir . '/php-ds/php-ds/src'),
@@ -60,2 +58,5 @@
     'Cake\\Cache\\' => array($vendorDir . '/cakephp/cache'),
+
+    'Monolog\\' => array($phpDir . '/Monolog'),
+    'Psr\\Log\\' => array($phpDir. '/Psr/Log'),
 );
--- eventum-3.6.3-29-g154295bd2/vendor/composer/autoload_real.php~	2019-03-26 09:59:16.000000000 +0200
+++ eventum-3.6.3-29-g154295bd2/vendor/composer/autoload_real.php	2019-03-26 13:18:14.402093309 +0200
@@ -23,7 +23,7 @@
         self::$loader = $loader = new \Composer\Autoload\ClassLoader();
         spl_autoload_unregister(array('ComposerAutoloaderInitEventumCore', 'loadClassLoader'));
 
-        $useStaticLoader = PHP_VERSION_ID >= 50600 && !defined('HHVM_VERSION') && (!function_exists('zend_loader_file_encoded') || !zend_loader_file_encoded());
+        $useStaticLoader = false; // autoload_static.php patching is annoying and not worth the effort
         if ($useStaticLoader) {
             require_once __DIR__ . '/autoload_static.php';
 
