--- eventum-3.6.7-73-ga12094613/globals.php	2019-05-17 00:29:37.284776088 +0300
+++ eventum-3.6.7-73-ga12094613/globals.php	2019-05-17 00:30:55.133655597 +0300
@@ -22,10 +22,8 @@
-// /var path for writable data
-define('APP_VAR_PATH', APP_PATH . '/var');
-define('APP_CACHE_PATH', APP_VAR_PATH . '/cache');
+define('APP_CACHE_PATH', '/var/cache/eventum');
 
 // define other paths
 define('APP_TPL_PATH', APP_PATH . '/templates');
 define('APP_TPL_COMPILE_PATH', APP_CACHE_PATH . '/smarty');
-define('APP_LOG_PATH', APP_VAR_PATH . '/log');
+define('APP_LOG_PATH', '/var/log/eventum');
 define('APP_ERROR_LOG', APP_LOG_PATH . '/errors.log');
-define('APP_LOCKS_PATH', APP_VAR_PATH . '/lock');
+define('APP_LOCKS_PATH', '/var/run/eventum');
--- eventum-3.5.6-98-g5b87d2b0/src/Setup/Requirements.php~	2019-01-23 01:50:42.000000000 +0200
+++ eventum-3.5.6-98-g5b87d2b0/src/Setup/Requirements.php	2019-01-24 22:18:56.047816296 +0200
@@ -64,10 +64,6 @@
         $configPath = Setup::getConfigPath();
         $setupFile = Setup::getSetupFile();
 
-        $error = $this->checkPermissions($configPath, "Directory '" . $configPath . "'", true);
-        if (!empty($error)) {
-            $errors[] = $error;
-        }
         $error = $this->checkPermissions($setupFile, "File '" . $setupFile . "'");
         if (!empty($error)) {
             $errors[] = $error;
--- eventum-3.5.0/src/Console/Command/MonitorCommand.php~	2018-05-20 17:33:34.000000000 +0300
+++ eventum-3.5.0/src/Console/Command/MonitorCommand.php	2018-05-21 01:42:21.463354763 +0300
@@ -47,17 +47,17 @@
         $required_files = [
             $configPath . '/config.php' => [
                 'check_owner' => true,
-                'owner' => 'apache',
+                'owner' => 'root',
                 'check_group' => true,
-                'group' => 'apache',
+                'group' => 'http',
                 'check_permission' => true,
                 'permission' => 640,
             ],
             $configPath . '/setup.php' => [
                 'check_owner' => true,
-                'owner' => 'apache',
+                'owner' => 'root',
                 'check_group' => true,
-                'group' => 'apache',
+                'group' => 'http',
                 'check_permission' => true,
                 'permission' => 660,
                 'check_filesize' => true,
--- eventum-3.1.5/config/sphinx.conf.php~	2016-11-13 12:05:37.000000000 +0200
+++ eventum-3.1.5/config/sphinx.conf.php	2016-11-23 03:10:05.855040510 +0200
@@ -1,7 +1,7 @@
 #!/usr/bin/php
 <?php
 
-require_once __DIR__ . '/../init.php';
+require_once '/usr/share/eventum/init.php';
 
 if (!defined('SPHINX_LOG_PATH')) {
     define('SPHINX_LOG_PATH', '/var/log/sphinx/');
--- eventum-3.3.1/src/Attachment/StorageManager.php~	2017-10-08 15:53:56.000000000 +0300
+++ eventum-3.3.1/src/Attachment/StorageManager.php	2017-10-09 23:22:00.162142738 +0300
@@ -32,7 +32,7 @@
      *
      * @var string
      */
-    const STORAGE_PATH = APP_PATH . '/var/storage/';
+    const STORAGE_PATH = '/var/lib/eventum/storage/';
 
     /**
      * @var MountManager
--- eventum-3.7.1/src/Kernel.php~	2019-04-06 17:42:28.000000000 +0300
+++ eventum-3.7.1/src/Kernel.php	2019-06-02 23:02:32.198743341 +0300
@@ -87,11 +87,11 @@
 
     public function getCacheDir(): string
     {
-        return "{$this->rootDir}/var/cache/{$this->environment}";
+        return APP_CACHE_PATH . '/' . $this->environment;
     }
 
     public function getLogDir(): string
     {
-        return "{$this->rootDir}/var/log";
+        return APP_LOG_PATH;
     }
 }
