--- eventum-1.7.1/scm_ping.php~	2006-06-19 17:42:09.926721975 +0300
+++ eventum-1.7.1/scm_ping.php	2006-06-19 17:44:06.729328243 +0300
@@ -26,7 +26,14 @@
 // +----------------------------------------------------------------------+
 //
 // @(#) $Id$
-//
+
+// shortcut to exit out when no issue id-s are passed in request
+// as this script is always called by CVS but we handle only ones which can be 
+// associated with issues.
+if (empty($_GET['issue']) {
+    exit;
+}
+
 include_once("config.inc.php");
 include_once(APP_INC_PATH . "class.misc.php");
 include_once(APP_INC_PATH . "class.scm.php");
