--- ./include/class.issue.php	2005-12-30 19:48:40.000000000 +0200
+++ ./include/class.issue.phps	2006-01-03 23:53:27.000000000 +0200
@@ -421,7 +421,7 @@
             Reminder_Action::clearLastTriggered($issue_id);
 
             // if old status was closed and new status is not, clear closed data from issue.
-            if ($old_details['sta_is_closed'] == 1) {
+            if (@$old_details['sta_is_closed'] == 1) {
                 $new_details = Status::getDetails($status_id);
                 if ($new_details['sta_is_closed'] != 1) {
                     Issue::clearClosed($issue_id);
@@ -1349,7 +1349,6 @@
                     $assignment_notifications[] = $assignee;
                     $assignments_changed = true;
                 }
-                $assignments_changed = true;
             }
             if (count($assignment_notifications) > 0) {
                 Notification::notifyNewAssignment($assignment_notifications, $issue_id);
@@ -2047,14 +2046,6 @@
                     }
                 }
             }
-            if (count($users) > 0) {
-                // automatically change the status to 'Assigned'
-                Issue::setStatus($new_issue_id, Status::getStatusID('Assigned'), FALSE);
-
-                // set this special variable to false, to avoid triggering
-                // another status update on the workflow class
-                $has_assignee = false;
-            }
 
             // now process any files being uploaded
             $found = 0;
