--- /usr/share/eventum/irc/bot.php	2005-05-17 23:03:42.000000000 +0300
+++ ./irc/eventum-irc-bot	2005-05-24 14:30:35.000000000 +0300
@@ -367,7 +367,7 @@
 $irc->setLogdestination(SMARTIRC_FILE);
 $irc->setLogfile(APP_IRC_LOG);
 $irc->setUseSockets(TRUE);
-$irc->setAutoReconnect(TRUE);
+$irc->setAutoReconnect(FALSE);
 $irc->setAutoRetry(TRUE);
 
 // register saytime() to be called every 30 sec. (30,000 milliseconds)
