--- eventum-1.7.1/include/class.mail.php~	2006-05-29 00:33:40.514188072 +0300
+++ eventum-1.7.1/include/class.mail.php	2006-05-29 00:34:52.765806904 +0300
@@ -571,7 +571,11 @@
         $to = MIME_Helper::encodeAddress($to);
         $subject = MIME_Helper::encode($subject);
 
-        $body = $this->mime->get();
+        $body = $this->mime->get(array(
+            'html_charset' => APP_CHARSET,
+            'text_charset' => APP_CHARSET,
+            'head_charset' => APP_CHARSET
+        ));
         $headers = array(
             'From'    => $from,
             'To'      => Mail_API::fixAddressQuoting($to),
@@ -612,7 +616,11 @@
         $to = MIME_Helper::encodeAddress($to);
         $subject = MIME_Helper::encode($subject);
 
-        $body = $this->mime->get();
+        $body = $this->mime->get(array(
+            'html_charset' => APP_CHARSET,
+            'text_charset' => APP_CHARSET,
+            'head_charset' => APP_CHARSET
+        ));
         $this->setHeaders(array(
             'From'    => $from,
             'To'      => $to,
