--- eventum-3.0.0/irc/eventum-irc-bot.php~	2015-02-02 20:07:00.097413069 +0200
+++ eventum-3.0.0/irc/eventum-irc-bot.php	2015-02-02 20:07:19.803776739 +0200
@@ -442,7 +442,7 @@
 $irc->setLogdestination(SMARTIRC_FILE);
 $irc->setLogfile(APP_IRC_LOG);
 $irc->setUseSockets(true);
-$irc->setAutoReconnect(true);
+$irc->setAutoReconnect(false);
 $irc->setAutoRetry(true);
 $irc->setReceiveTimeout(600);
 $irc->setTransmitTimeout(600);
