--- eventum/misc/irc/bot.php~	2007-02-09 00:55:47.199440664 +0200
+++ eventum/misc/irc/bot.php	2007-02-09 01:39:53.609120026 +0200
@@ -1,3 +1,4 @@
+#!/usr/bin/php
 <?php
 /* vim: set expandtab tabstop=4 shiftwidth=4 encoding=utf-8: */
 // +----------------------------------------------------------------------+
@@ -30,7 +31,7 @@
 
 ini_set('memory_limit', '1024M');
 
-require_once(dirname(__FILE__) . '/../../init.php');
+require_once('/usr/share/eventum/htdocs/init.php');
 
 if (!file_exists(APP_CONFIG_PATH . 'irc_config.php')) {
     echo "ERROR: No config specified. Please see setup/irc_config.php for config information.\n\n";
