--- eventum-r3749/js/global.js~	2008-10-15 01:57:31.000000000 +0300
+++ eventum-r3749/js/global.js	2008-10-15 01:57:53.000000000 +0300
@@ -794,7 +794,7 @@
 // call when document ready
 $(document).ready(function() {
     $('.date_picker').datepicker({
-        dateFormat: 'yy-mm-dd', firstDay: 0  // FIXME: make it configurable
+        dateFormat: 'yy-mm-dd', firstDay: 1  // FIXME: make it configurable
     });
 }
 
