--- eventum-1.4/setup/schema.sql~	2005-01-05 01:06:07.000000000 +0200
+++ eventum-1.4/setup/schema.sql	2005-01-24 13:45:36.000000000 +0200
@@ -108,6 +108,7 @@
 INSERT INTO %TABLE_PREFIX%history_type SET htt_name = 'phone_entry_added', htt_role = 4;
 INSERT INTO %TABLE_PREFIX%history_type SET htt_name = 'phone_entry_removed', htt_role = 4;
 INSERT INTO %TABLE_PREFIX%history_type SET htt_name = 'scm_checkin_removed';
+INSERT INTO %TABLE_PREFIX%history_type SET htt_name = 'scm_checkin_associated';
 INSERT INTO %TABLE_PREFIX%history_type SET htt_name = 'email_associated';
 INSERT INTO %TABLE_PREFIX%history_type SET htt_name = 'email_disassociated';
 INSERT INTO %TABLE_PREFIX%history_type SET htt_name = 'email_sent';
