diff -urN eventum.backup.4/stats_chart.php eventum/stats_chart.php
--- eventum.backup.4/stats_chart.php	2007-03-06 15:57:48.000000000 +0100
+++ eventum/stats_chart.php	2007-03-07 10:29:54.000000000 +0100
@@ -71,7 +71,7 @@
     $data = Stats::getAssocCategory();
     $graph_title = ev_gettext("Issues by Category");
 }
-$labels = array_map('utf8_decode', array_keys($data));
+$labels = array_keys($data);
 $data = array_values($data);
 
 // check the values coming from the database and if they are all empty, then
