--- eventum-3.7.4-440-gcc5ad3db5/./src/Translation.php~	2019-08-12 07:40:03.000000000 +0300
+++ eventum-3.7.4-440-gcc5ad3db5/./src/Translation.php	2019-08-21 23:35:06.911034648 +0300
@@ -19,7 +19,6 @@
 
     public static function init(string $domain, string $path = null): void
     {
-        bindtextdomain($domain, $path);
         bind_textdomain_codeset($domain, self::CHARSET);
         textdomain($domain);
     }
