--- eventum/misc/scm/process_cvs_commits.php~	2007-02-15 01:28:29.670480306 +0200
+++ eventum/misc/scm/process_cvs_commits.php	2007-02-15 01:28:52.320991777 +0200
@@ -28,15 +28,7 @@
 //
 // @(#) $Id$
 
-
-// URL to your Eventum installation.
-// https is supported transparently by PHP 5 if you have openssl module enabled.
-$eventum_url = 'http://rabbit.impleo.net/';
-
-
-//
-// DO NOT CHANGE ANYTHING AFTER THIS LINE
-//
+include '/etc/eventum/scm.php';
 
 if (isset($eventum_url)) {
     $data = parse_url($eventum_url);
