--- ./send.php~	2005-02-27 17:59:14.000000000 +0200
+++ ./send.php	2005-03-04 16:16:05.000000000 +0200
@@ -134,6 +134,11 @@
             "email"           => $email,
             "parent_email_id" => $HTTP_GET_VARS["id"]
         ));
+    } else {
+        $email["reply_subject"] = "[#" . @$HTTP_GET_VARS['issue_id'] . "] ";
+        $tpl->bulkAssign(array(
+            "email"           => $email,
+        ));
     }
 }
 
