--- ./misc/irc/bot.php.bak	2005-08-20 05:28:05.000000000 +0300
+++ ./misc/irc/bot.php	2005-08-20 05:33:35.000000000 +0300
@@ -28,7 +28,6 @@
 // @(#) $Id$
 //
 
-include_once("../../config.inc.php");
 include_once(APP_INC_PATH . "db_access.php");
 include_once(APP_INC_PATH . "class.auth.php");
 include_once(APP_INC_PATH . "class.lock.php");
@@ -41,8 +40,8 @@
 // the following is the list of IRC channels that the bot should connect to,
 // and the associated project name
 $channels = array(
-    Project::getID('Default Project') => array(
-        '#issues',
+    Project::getID($irc_default_project) => array(
+        $irc_default_channel,
     )
 );
 $irc_server_hostname = 'localhost';
@@ -54,6 +53,8 @@
 $username = '';
 $password = '';
 
+include_once('/etc/eventum/irc.php');
+
 
 // ============================================
 // ============================================
