--- ./lib/eventum/class.monitor.php~	2005-02-22 21:50:49.000000000 +0200
+++ ./lib/eventum/class.monitor.php	2005-02-22 23:04:50.000000000 +0200
@@ -157,6 +165,7 @@
                 'permission'       => 100,
             ),
         );
+        $required_directories = array();
         foreach ($required_directories as $dir_path => $options) {
             // check if directory exists
             if (!file_exists($dir_path)) {
--- ./cli/eventum	2009-06-27 19:46:44.401524716 +0300
+++ ./cli/eventum	2009-06-27 19:45:22.781438493 +0300
@@ -12,9 +12,8 @@
 // @(#) $Id$
 
 // definitions of path related variables
-define('APP_PATH', dirname(__FILE__) . '/');
-define('APP_INC_PATH', APP_PATH . '/lib/eventum');
-define('APP_PEAR_PATH', APP_PATH . '/lib/pear');
+define('APP_PATH', '/usr/share/eventum');
+define('APP_INC_PATH', APP_PATH . '/cli');
 
 if (defined('APP_PEAR_PATH')) {
     set_include_path(APP_PEAR_PATH . PATH_SEPARATOR . get_include_path());
--- eventum-2.2/init.php~	2009-10-12 22:40:04.000000000 +0300
+++ eventum-2.2/init.php	2009-10-12 22:40:41.619219457 +0300
@@ -27,7 +27,7 @@
 // | Authors: Elan Ruusamäe <glen@delfi.ee>                               |
 // +----------------------------------------------------------------------+
 
-if (!file_exists(dirname(__FILE__) . '/config/config.php') || !filesize(dirname(__FILE__) . '/config/config.php')) {
+if (!file_exists('/etc/webapps/eventum/config.php') || !filesize('/etc/webapps/eventum/config.php')) {
     Header('Location: setup/');
     exit(0);
 }
@@ -46,7 +46,7 @@
 
 // define base path
 define('APP_PATH', realpath(dirname(__FILE__)));
-define('APP_CONFIG_PATH', APP_PATH . '/config');
+define('APP_CONFIG_PATH', '/etc/webapps/eventum');
 
 // include local site config. may override any default
 require_once APP_CONFIG_PATH . '/config.php';
@@ -61,19 +61,15 @@
 }
 
 if (!defined('APP_TPL_COMPILE_PATH')) {
-    define('APP_TPL_COMPILE_PATH', APP_PATH . '/templates_c');
+    define('APP_TPL_COMPILE_PATH', '/var/cache/eventum');
 }
 
 if (!defined('APP_INC_PATH')) {
     define('APP_INC_PATH', APP_PATH . '/lib/eventum');
 }
 
-if (!defined('APP_PEAR_PATH')) {
-    define('APP_PEAR_PATH', APP_PATH . '/lib/pear');
-}
-
 if (!defined('APP_SMARTY_PATH')) {
-    define('APP_SMARTY_PATH', APP_PATH . '/lib/Smarty');
+    define('APP_SMARTY_PATH', '/usr/share/php/Smarty');
 }
 
 if (!defined('APP_JPGRAPH_PATH')) {
@@ -81,7 +77,7 @@
 }
 
 if (!defined('APP_LOCKS_PATH')) {
-    define('APP_LOCKS_PATH', APP_PATH . '/locks');
+    define('APP_LOCKS_PATH', '/var/run/eventum/');
 }
 
 if (!defined('APP_SQL_PATCHES_PATH')) {
@@ -89,7 +85,7 @@
 }
 
 if (!defined('APP_LOG_PATH')) {
-    define('APP_LOG_PATH', APP_PATH . '/logs');
+    define('APP_LOG_PATH', '/var/log/eventum');
 }
 
 if (!defined('APP_ROUTED_MAILS_SAVEDIR')) {
--- eventum-2.2/htdocs/setup/index.php~	2010-07-28 12:09:26.000000000 +0300
+++ eventum-2.2/htdocs/setup/index.php	2010-07-28 12:09:48.715993083 +0300
@@ -42,14 +42,14 @@
 define('APP_PATH', realpath(dirname(__FILE__) . '/../..'));
 define('APP_INC_PATH', APP_PATH . '/lib/eventum');
 define('APP_PEAR_PATH', APP_PATH . '/lib/pear');
-define('APP_SMARTY_PATH', APP_PATH . '/lib/Smarty');
-define('APP_CONFIG_PATH', APP_PATH . '/config');
+define('APP_SMARTY_PATH', '/usr/share/php/Smarty');
+define('APP_CONFIG_PATH', '/etc/webapps/eventum');
 define('APP_SETUP_FILE', APP_CONFIG_PATH . '/setup.php');
 define('APP_TPL_PATH', APP_PATH . '/templates');
-define('APP_TPL_COMPILE_PATH', APP_PATH . '/templates_c');
-define('APP_LOG_PATH', APP_PATH . '/logs');
+define('APP_TPL_COMPILE_PATH', '/var/cache/eventum');
+define('APP_LOG_PATH', '/var/log/eventum');
 define('APP_ERROR_LOG', APP_LOG_PATH . '/errors.log');
-define('APP_LOCKS_PATH', APP_PATH . '/locks');
+define('APP_LOCKS_PATH', '/var/run/eventum');
 
 header('Content-Type: text/html; charset=' . APP_CHARSET);
 
@@ -252,10 +252,6 @@
             "You need to install this extension for optimal operation. If you do not install this extension some unicode data will be corrupted.";
     }
 
-    $error = checkPermissions(APP_CONFIG_PATH, "Directory '" . APP_CONFIG_PATH . "'", TRUE);
-    if (!empty($error)) {
-        $errors[] = $error;
-    }
     $error = checkPermissions(APP_SETUP_FILE, "File '" . APP_SETUP_FILE. "'");
     if (!empty($error)) {
         $errors[] = $error;
@@ -329,10 +325,6 @@
     $setup_file_path = APP_SETUP_FILE;
 
     clearstatcache();
-    // check if config directory is writable
-    if (!is_writable(APP_CONFIG_PATH)) {
-        return "The file '" . APP_CONFIG_PATH . "' directory needs to be writable by the web server user. Please correct this problem and try again.";
-    }
     // need to create a random private key variable
     $private_key = '<?php
 $private_key = "' . md5(microtime()) . '";
--- eventum-2.2/irc/eventum-irc-bot~	2009-07-23 14:03:41.422211655 +0300
+++ eventum-2.2/irc/eventum-irc-bot	2009-07-23 13:54:22.000000000 +0300
@@ -32,7 +32,7 @@
 
 ini_set('memory_limit', '1024M');
 
-require_once dirname(__FILE__) . '/../init.php';
+require_once '/usr/share/eventum/init.php';
 
 if (!file_exists(APP_CONFIG_PATH . '/irc_config.php')) {
     fwrite(STDERR, "ERROR: No config specified. Please see setup/irc_config.php for config information.\n\n");
