--- ./rpc/xmlrpc.php~	2005-01-05 01:06:19.000000000 +0200
+++ ./rpc/xmlrpc.php	2005-01-25 16:35:14.000000000 +0200
@@ -41,7 +41,7 @@
 require_once(APP_INC_PATH . "class.template.php");
 require_once(APP_INC_PATH . "class.customer.php");
 error_reporting(0);
-require_once(APP_PEAR_PATH . "XML_RPC/Server.php");
+require_once(APP_PEAR_PATH . "XML/RPC/Server.php");
 
 function authenticate($email, $password)
 {
--- ./include/class.notification.php~	2005-01-05 01:06:19.000000000 +0200
+++ ./include/class.notification.php	2005-01-25 16:41:23.000000000 +0200
@@ -613,9 +613,9 @@
         }
         if ($old["iss_description"] != $new["description"]) {
             // need real diff engine here
-            require_once 'Text_Diff/Diff.php';
-            require_once 'Text_Diff/Diff/Renderer.php';
-            require_once 'Text_Diff/Diff/Renderer/unified.php';
+            require_once 'Text/Diff.php';
+            require_once 'Text/Diff/Renderer.php';
+            require_once 'Text/Diff/Renderer/unified.php';
             $old['iss_description'] = explode("\n", $old['iss_description']);
             $new['description'] = explode("\n", $new['description']);
             $diff = &new Text_Diff($old["iss_description"], $new["description"]);
--- eventum/misc/cli/eventum~	2007-01-30 20:29:08.299995813 +0200
+++ eventum/misc/cli/eventum	2007-01-30 20:29:37.750658504 +0200
@@ -14,5 +14,5 @@
 require_once(dirname(__FILE__) . '/config.inc.php');
 require_once(APP_INC_PATH . 'class.command_line.php');
-require_once(APP_PEAR_PATH . 'XML_RPC/RPC.php');
+require_once(APP_PEAR_PATH . 'XML/RPC.php');
 
 list($user_email, $user_password, $url, $port, $relative_url) = Command_Line::getEnvironmentSettings();
--- eventum/misc/cli/include/class.command_line.php~	2007-01-30 20:29:08.299995813 +0200
+++ eventum/misc/cli/include/class.command_line.php	2007-01-30 20:31:39.333394375 +0200
@@ -30,7 +30,7 @@
 
 require_once(APP_INC_PATH . 'class.misc.php');
 require_once(APP_PEAR_PATH . 'PEAR.php');
-require_once(APP_PEAR_PATH . 'XML_RPC/RPC.php');
+require_once(APP_PEAR_PATH . 'XML/RPC.php');
 
 $_displayed_confirmation = false;
 
