--- eventum-1.4.c/setup/config.inc.php	2005-01-05 01:05:58.000000000 +0200
+++ eventum-1.4/setup/config.inc.php	2005-01-24 17:21:52.000000000 +0200
@@ -56,10 +56,10 @@
 @define("APP_INC_PATH", APP_PATH . "include/");
 @define("APP_PEAR_PATH", APP_INC_PATH . "pear/");
 @define("APP_TPL_PATH", APP_PATH . "templates/");
-@define("APP_SMARTY_PATH", APP_INC_PATH . "Smarty/");
+@define("APP_SMARTY_PATH", "/usr/share/pear/Smarty/");
 @define("APP_JPGRAPH_PATH", APP_INC_PATH . "jpgraph/");
-@define("APP_LOG_PATH", APP_PATH . "logs/");
-@define("APP_LOCKS_PATH", APP_PATH . "locks/");
+@define("APP_LOG_PATH", "/var/log/eventum/");
+@define("APP_LOCKS_PATH", "/var/run/eventum/");
 if (stristr(PHP_OS, 'darwin')) {
     ini_set("include_path", ".:" . APP_PEAR_PATH);
 } elseif (stristr(PHP_OS, 'win')) {
@@ -153,4 +153,4 @@
 
 // set charset
 header("content-type: text/html;charset=" . APP_CHARSET);
-?>
\ No newline at end of file
+?>
--- ./misc/cli/eventum~	2005-01-19 02:47:20.000000000 +0200
+++ ./misc/cli/eventum	2005-01-19 03:02:40.119100405 +0200
@@ -11,7 +11,7 @@
 //
 // @(#) $Id$
 //
-include_once("config.inc.php");
+include_once("/etc/eventum/cli.php");
 include_once(APP_INC_PATH . "class.command_line.php");
 include_once(APP_PEAR_PATH . "XML_RPC/RPC.php");
 
--- eventum-1.4/misc/cli/config.inc.php~	2005-01-05 01:06:01.000000000 +0200
+++ eventum-1.4/misc/cli/config.inc.php	2005-01-19 03:25:10.000000000 +0200
@@ -45,7 +45,7 @@
 }
 
 // definitions of path related variables
-@define("APP_PATH", dirname(__FILE__) . '/');
+@define("APP_PATH", '/usr/share/eventum/misc/cli/');
 @define("APP_INC_PATH", APP_PATH . "include/");
 @define("APP_PEAR_PATH", APP_INC_PATH . "pear/");
 if (stristr(PHP_OS, 'darwin')) {
@@ -57,4 +57,4 @@
 }
 
 @define("APP_BENCHMARK", false);
-?>
\ No newline at end of file
+?>
--- eventum-1.4.c/include/class.template.php	2005-01-05 01:05:59.000000000 +0200
+++ eventum-1.4/include/class.template.php	2005-01-24 17:22:35.000000000 +0200
@@ -60,7 +60,7 @@
     {
         $this->smarty = new Smarty;
         $this->smarty->template_dir = APP_PATH . "templates/" . APP_CURRENT_LANG;
-        $this->smarty->compile_dir = APP_PATH . "templates_c";
+        $this->smarty->compile_dir = "/var/cache/eventum";
         $this->smarty->config_dir = '';
         $this->smarty->register_modifier("activateLinks", array('Link_Filter', 'processText'));
     }
--- index.php	2005-01-05 01:05:52.000000000 +0200
+++ eventum-1.4/setup/index.php	2005-01-24 17:47:16.000000000 +0200
@@ -114,15 +114,15 @@
     if (ini_get('allow_call_time_pass_reference') != "1") {
         $errors[] = "The 'allow_call_time_pass_reference' directive needs to be enabled in your PHP.INI file in order for Eventum to work properly.";
     }
-    $error = checkPermissions('../locks', "Directory 'locks'", TRUE);
+    $error = checkPermissions('/var/run/eventum', "Directory '/var/run/eventum'", TRUE);
     if (!empty($error)) {
         $errors[] = $error;
     }
-    $error = checkPermissions('../logs', "Directory 'logs'", TRUE);
+    $error = checkPermissions('/var/log/eventum', "Directory '/var/log/eventum'", TRUE);
     if (!empty($error)) {
         $errors[] = $error;
     }
-    $error = checkPermissions('../templates_c', "Directory 'templates_c'", TRUE);
+    $error = checkPermissions('/var/cache/eventum', "Directory '/var/cache/eventum'", TRUE);
     if (!empty($error)) {
         $errors[] = $error;
     }
@@ -130,7 +130,7 @@
     if (!empty($error)) {
         $errors[] = $error;
     }
-    $error = checkPermissions('../logs/errors.log', "File 'logs/errors.log'");
+    $error = checkPermissions('/var/log/eventum/errors.log', "File '/var/log/eventum/errors.log'");
     if (!empty($error)) {
         $errors[] = $error;
     }
@@ -199,11 +199,11 @@
 }
 
 ini_set("include_path", '.');
-include_once("../include/Smarty/Smarty.class.php");
+include_once('/usr/share/pear/Smarty/Smarty.class.php');
 
 $tpl = new Smarty();
 $tpl->template_dir = '../templates/en';
-$tpl->compile_dir = "../templates_c";
+$tpl->compile_dir = "/var/cache/eventum";
 $tpl->config_dir = '';
 
 function replace_table_prefix($str)
@@ -417,23 +417,8 @@
     $tpl->assign('is_imap_enabled', function_exists('imap_open'));
 }
 
-
-$full_url = dirname($HTTP_SERVER_VARS['PHP_SELF']);
-$pieces = explode("/", $full_url);
-$relative_url = array();
-$relative_url[] = '';
-foreach ($pieces as $piece) {
-    if ((!empty($piece)) && ($piece != 'setup')) {
-        $relative_url[] = $piece;
-    }
-}
-$relative_url[] = '';
-$relative_url = implode("/", $relative_url);
-
-if (substr($HTTP_SERVER_VARS['DOCUMENT_ROOT'], -1) == '/') {
-    $HTTP_SERVER_VARS['DOCUMENT_ROOT'] = substr($HTTP_SERVER_VARS['DOCUMENT_ROOT'], 0, -1);
-}
-$installation_path = $HTTP_SERVER_VARS['DOCUMENT_ROOT'] . $relative_url;
+$installation_path = "/usr/share/eventum";
+$relative_url = "/eventum/";
 
 $tpl->assign("phpversion", phpversion());
 $tpl->assign("rel_url", $relative_url);
--- misc/cli/eventum~	2005-01-24 17:49:25.000000000 +0200
+++ ./misc/cli/eventum	2005-01-25 15:48:56.000000000 +0200
@@ -13,7 +13,7 @@
 //
 include_once("/etc/eventum/cli.php");
 include_once(APP_INC_PATH . "class.command_line.php");
-include_once(APP_PEAR_PATH . "XML_RPC/RPC.php");
+include_once('/usr/share/pear/XML/RPC.php');
 
 list($user_email, $user_password, $url, $port, $relative_url) = Command_Line::getEnvironmentSettings();
 if (empty($port)) {
--- misc/cli/config.inc.php~	2005-01-24 17:49:25.000000000 +0200
+++ ./misc/cli/config.inc.php	2005-01-25 15:52:07.000000000 +0200
@@ -47,7 +47,7 @@
 // definitions of path related variables
 @define("APP_PATH", '/usr/share/eventum/misc/cli/');
 @define("APP_INC_PATH", APP_PATH . "include/");
-@define("APP_PEAR_PATH", APP_INC_PATH . "pear/");
+@define("APP_PEAR_PATH", "/usr/share/pear/");
 if (stristr(PHP_OS, 'darwin')) {
     ini_set("include_path", ".:" . APP_PEAR_PATH);
 } elseif (stristr(PHP_OS, 'win')) {
--- misc/cli/include/class.command_line.php	2005-01-05 01:06:03.000000000 +0200
+++ ./misc/cli/include/class.command_line.php	2005-01-25 15:48:56.000000000 +0200
@@ -29,7 +29,7 @@
 //
 
 include_once(APP_INC_PATH . "class.misc.php");
-include_once(APP_PEAR_PATH . "XML_RPC/RPC.php");
+include_once('/usr/share/pear/XML/RPC.php');
 
 $_displayed_confirmation = false;
 
