--- eventum-20060830/include/class.draft.php~	2006-08-30 20:27:26.808707134 +0300
+++ eventum-20060830/include/class.draft.php	2006-08-31 12:35:25.201965444 +0300
@@ -139,7 +139,7 @@
     {
         $issue_id = Misc::escapeInteger($issue_id);
         $emd_id = Misc::escapeInteger($emd_id);
-        $parent_id = Misc::escapeInteger($issue_id);
+        $parent_id = Misc::escapeInteger($parent_id);
         if (empty($parent_id)) {
             $parent_id = 'NULL';
         }
