--- eventum-r3475/lib/eventum/class.language.php~	2007-11-20 19:11:35.000000000 +0200
+++ eventum-r3475/lib/eventum/class.language.php	2007-11-20 19:12:05.696195147 +0200
@@ -237,7 +237,6 @@
 }
 
 // this won't change over the request. so set it once and permanently
-_bindtextdomain('eventum', APP_PATH . 'misc/localization/');
 _bind_textdomain_codeset('eventum', APP_CHARSET);
 _textdomain('eventum');
 
