--- eventum-2.3.2/scm/eventum-cvs-hook.php~	2012-02-09 11:27:05.000000000 +0200
+++ eventum-2.3.2/scm/eventum-cvs-hook.php	2012-02-09 11:29:22.671597765 +0200
@@ -28,13 +28,7 @@
 // |          Elan Ruusamäe <glen@delfi.ee>                               |
 // +----------------------------------------------------------------------+
 
-// URL to your Eventum installation.
-// https is supported transparently by PHP 5 if you have openssl module enabled.
-$eventum_url = 'http://eventum.example.com/';
-
-//
-// DO NOT CHANGE ANYTHING AFTER THIS LINE
-//
+require_once '/etc/eventum/scm.php';
 
 // save name of this script
 $PROGRAM = basename(array_shift($argv));
--- eventum-2.2/scm/eventum-svn-hook.php~	2009-11-10 13:31:10.000000000 +0200
+++ eventum-2.2/scm/eventum-svn-hook.php	2009-11-10 13:31:37.571582772 +0200
@@ -36,7 +36,7 @@
 
 // URL to your Eventum installation.
 // https is supported transparently by PHP 5 if you have openssl module enabled.
-$eventum_url = 'http://eventum.example.com/';
+require_once '/etc/eventum/scm.php';
 
 //
 // DO NOT CHANGE ANYTHING AFTER THIS LINE
