--- ./misc/download_emails.php	2005-10-25 00:46:28.000000000 +0300
+++ /tmp/download_emails.php	2005-10-25 00:53:15.000000000 +0300
@@ -88,7 +88,7 @@
 }
 
 // get the account ID since we need it for locking.
-$account_id = Email_Account::getAccountID($username, $hostname, $mailbox);
+$account_id = @Email_Account::getAccountID($username, $hostname, $mailbox);
 if (($account_id == 0) && ($fix_lock != true)) {
     echo "Error: Could not find a email account with the parameter provided. Please verify your email account settings and try again.\n";
     exit;
@@ -127,7 +127,7 @@
     exit;
 }
 
-$account = Email_Account::getDetails($account_id);
+$account = @Email_Account::getDetails($account_id);
 $mbox = Support::connectEmailServer($account);
 if ($mbox == false) {
     echo "Error: Could not connect to the email server. Please verify your email account settings and try again.\n";
@@ -137,7 +137,7 @@
     $total_emails = Support::getTotalEmails($mbox);
     if ($total_emails > 0) {
         for ($i = 1; $i <= $total_emails; $i++) {
-            Support::getEmailInfo($mbox, $account, $i);
+            @Support::getEmailInfo($mbox, $account, $i);
         }
     }
     imap_expunge($mbox);
--- ./misc/process_mail_queue.php	2005-10-25 00:56:14.000000000 +0300
+++ /tmp/process_mail_queue.php	2005-10-25 01:20:51.000000000 +0300
@@ -63,11 +63,11 @@
 
 // handle only pending emails
 $limit = 50;
-Mail_Queue::send('pending', $limit);
+@Mail_Queue::send('pending', $limit);
 
 // handle emails that we tried to send before, but an error happened...
 $limit = 50;
-Mail_Queue::send('error', $limit);
+@Mail_Queue::send('error', $limit);
 
 Mail_Queue::removeProcessFile();
-?>
\ No newline at end of file
+?>
