--- eventum-1.7.1/include/class.support.php~	2006-07-15 21:11:21.020205702 +0300
+++ eventum-1.7.1/include/class.support.php	2006-07-15 21:13:36.033229467 +0300
@@ -430,11 +430,16 @@
      */
     function connectEmailServer($info)
     {
-        $mbox = @imap_open(Support::getServerURI($info), $info['ema_username'], $info['ema_password']);
+        $username = $info['ema_username'];
+        $password = $info['ema_password'];
+        # avoid password showing up in backtrace
+        unset($info['ema_password']);
+
+        $mbox = @imap_open(Support::getServerURI($info), $username, $password);
         if ($mbox === FALSE) {
             $errors = @imap_errors();
             if (strstr(strtolower($errors[0]), 'certificate failure')) {
-                $mbox = @imap_open(Support::getServerURI($info, TRUE), $info['ema_username'], $info['ema_password']);
+                $mbox = @imap_open(Support::getServerURI($info, TRUE), $username, $password);
             } else {
                 Error_Handler::logError('Error while connecting to the email server - ' . $errors[0], __FILE__, __LINE__);
             }
