--- eventum-1.7.0/misc/irc/bot.php~	2005-12-29 21:27:25.000000000 +0200
+++ eventum-1.7.0/misc/irc/bot.php	2005-12-30 00:14:31.000000000 +0200
@@ -28,6 +28,8 @@
 // @(#) $Id$
 //
 
+ini_set("memory_limit", "256M");
+
 include_once("../../config.inc.php");
 include_once(APP_INC_PATH . "db_access.php");
 include_once(APP_INC_PATH . "class.auth.php");
@@ -36,8 +38,6 @@
 include_once(APP_INC_PATH . "class.user.php");
 include_once(APP_PEAR_PATH . 'Net/SmartIRC.php');
 
-ini_set("memory_limit", "256M");
-
 // the following is the list of IRC channels that the bot should connect to,
 // and the associated project name
 $channels = array(
