--- eventum-r3374/misc/upgrade/v2.0_to_v2.1/database_changes.php~	2007-09-06 12:10:01.000000000 +0300
+++ eventum-r3374/misc/upgrade/v2.0_to_v2.1/database_changes.php	2007-09-06 12:10:29.649982963 +0300
@@ -7,9 +7,6 @@
 $stmts[] = "ALTER TABLE eventum_irc_notice ADD INDEX ino_status (ino_status)";
 $stmts[] = "ALTER TABLE eventum_issue_custom_field ADD COLUMN icf_value_integer int(11) NULL DEFAULT NULL";
 $stmts[] = "ALTER TABLE eventum_issue_custom_field ADD COLUMN icf_value_date date NULL DEFAULT NULL";
-$stmts[] = "ALTER TABLE eventum_custom_field ADD COLUMN fld_close_form tinyint(1) NOT NULL DEFAULT 0";
-$stmts[] = "ALTER TABLE eventum_custom_field ADD COLUMN fld_close_form_required tinyint(1) NOT NULL DEFAULT 0";
-$stmts[] = "ALTER TABLE eventum_issue ADD COLUMN iss_customer_contract_id varchar(50) NULL AFTER iss_customer_contact_id";
 
 foreach ($stmts as $stmt) {
     $stmt = str_replace('eventum_', APP_TABLE_PREFIX, $stmt);
--- /dev/null	2006-03-28 14:00:37.387408000 +0300
+++ eventum-r3374/misc/upgrade/v2.0_to_v2.1/database_changes2.php	2007-09-06 12:10:21.879805496 +0300
@@ -0,0 +1,19 @@
+<?php
+require_once '/usr/share/eventum/htdocs/init.php';
+require_once(APP_INC_PATH . "db_access.php");
+
+$stmts = array();
+
+$stmts[] = "ALTER TABLE eventum_custom_field ADD COLUMN fld_close_form tinyint(1) NOT NULL DEFAULT 0";
+$stmts[] = "ALTER TABLE eventum_custom_field ADD COLUMN fld_close_form_required tinyint(1) NOT NULL DEFAULT 0";
+
+foreach ($stmts as $stmt) {
+    $stmt = str_replace('eventum_', APP_TABLE_PREFIX, $stmt);
+    $res = $GLOBALS["db_api"]->dbh->query($stmt);
+    if (PEAR::isError($res)) {
+        echo "<pre>";var_dump($res);echo "</pre>";
+        exit(1);
+    }
+}
+?>
+done
--- /dev/null	2006-03-28 14:00:37.000000000 +0300
+++ eventum-r3374/misc/upgrade/v2.0_to_v2.1/database_changes3.php	2007-11-04 17:40:19.385617634 +0200
@@ -0,0 +1,18 @@
+<?php
+require_once '/usr/share/eventum/htdocs/init.php';
+require_once(APP_INC_PATH . "db_access.php");
+
+$stmts = array();
+
+$stmts[] = "ALTER TABLE eventum_issue ADD COLUMN iss_customer_contract_id int(11) unsigned AFTER iss_customer_id";
+
+foreach ($stmts as $stmt) {
+    $stmt = str_replace('eventum_', APP_TABLE_PREFIX, $stmt);
+    $res = $GLOBALS["db_api"]->dbh->query($stmt);
+    if (PEAR::isError($res)) {
+        echo "<pre>";var_dump($res);echo "</pre>";
+        exit(1);
+    }
+}
+?>
+done
--- /dev/null	2006-03-28 14:00:37.000000000 +0300
+++ eventum-r3374/misc/upgrade/v2.0_to_v2.1/database_changes4.php	2007-11-04 17:40:19.385617634 +0200
@@ -0,0 +1,18 @@
+<?php
+require_once '/usr/share/eventum/htdocs/init.php';
+require_once(APP_INC_PATH . "db_access.php");
+
+$stmts = array();
+
+$stmts[] = "ALTER TABLE eventum_issue MODIFY COLUMN iss_customer_contract_id varchar(50) NULL";
+
+foreach ($stmts as $stmt) {
+    $stmt = str_replace('eventum_', APP_TABLE_PREFIX, $stmt);
+    $res = $GLOBALS["db_api"]->dbh->query($stmt);
+    if (PEAR::isError($res)) {
+        echo "<pre>";var_dump($res);echo "</pre>";
+        exit(1);
+    }
+}
+?>
+done
