-------- Original Message --------
Subject: bk commit - eventum-internal (bryan:1.1521)
Date: Tue, 13 Sep 2005 03:57:17 +0200
From: bryan

Remove 'double decoding' of emails.

--- 1.49/include/class.mime_helper.php	2005-08-09 18:11:27 +02:00
+++ 1.50/include/class.mime_helper.php	2005-09-13 03:56:49 +02:00
@@ -100,9 +100,6 @@
             $str = str_replace("</DIV><DIV>", "\n", $str);
             $str = str_replace(array("<br>", "<br />", "<BR>", "<BR />"), "\n", $str);
         }
-        if (@$output->headers['content-transfer-encoding'] == 'quoted-printable') {
-            $str = Mime_Helper::decodeBody($str, 'quoted-printable');
-        }
         // XXX: do we also need to do something here about base64 encoding?
         if ($is_html) {
             $str = strip_tags($str);
@@ -657,35 +657,6 @@
 
 
     /**
-     * Method used to decode the body of a MIME encoded message.
-     *
-     * @access  public
-     * @param   string $input The full body of the message
-     * @param   string $encoding The encoding used in the message
-     * @return  string The decoded message body
-     */
-    function decodeBody($input, $encoding = '7bit')
-    {
-        switch ($encoding) {
-            case '7bit':
-                return $input;
-                break;
-
-            case 'quoted-printable':
-                return Mime_Helper::_quotedPrintableDecode($input);
-                break;
-
-            case 'base64':
-                return base64_decode($input);
-                break;
-
-            default:
-                return $input;
-        }
-    }
-
-
-    /**
      * Given a quoted-printable string, this
      * function will decode and return it.
      *
