--- include/class.monitor.php	2005-02-24 00:15:00.000000000 +0200
+++ ./include/class.monitor.php	2005-02-24 12:17:51.000000000 +0200
@@ -290,7 +290,7 @@
     {
         // check if any bot.php process is still running (lame, but oh well)
         ob_start();
-        passthru("ps -ef | grep 'php -q bot.php'");
+        passthru("ps -ef | grep '[-]q bot.php'");
         $contents = ob_get_contents();
         ob_end_clean();
         $lines = explode("\n", $contents);
