--- /tmp/process_cvs_commits.php	2005-01-17 20:51:29.343450373 +0200
+++ ./misc/scm/process_cvs_commits.php	2005-01-17 20:51:10.639892676 +0200
@@ -74,13 +78,17 @@
 if (count($matches) > 1) {
     $issue_id = $matches[1];
     $commit_msg = base64_encode($commit_msg);
+    $cvs_module = base64_encode($cvs_module);
+    $username = base64_encode($username);
+    $issue_id = base64_encode($issue_id);
+
     // build the GET url to use
     $ping_url = $eventum_relative_url . "scm_ping.php?module=$cvs_module&username=$username&commit_msg=$commit_msg";
     $ping_url .= "&issue[]=$issue_id";
     for ($i = 0; $i < count($modified_files); $i++) {
-        $ping_url .= "&files[$i]=" . $modified_files[$i]['filename'];
-        $ping_url .= "&old_versions[$i]=" . $modified_files[$i]['old_revision'];
-        $ping_url .= "&new_versions[$i]=" . $modified_files[$i]['new_revision'];
+        $ping_url .= "&files[$i]=" . base64_encode($modified_files[$i]['filename']);
+        $ping_url .= "&old_versions[$i]=" . base64_encode($modified_files[$i]['old_revision']);
+        $ping_url .= "&new_versions[$i]=" . base64_encode($modified_files[$i]['new_revision']);
     }
 
     $fp = fsockopen($eventum_domain, $eventum_port, $errno, $errstr, 30);
