--- eventum-1.5.1/misc/irc/bot.php~	2005-03-28 23:30:33.000000000 +0300
+++ eventum-1.5.1/misc/irc/bot.php	2005-03-29 00:03:57.000000000 +0300
@@ -36,6 +36,8 @@
 include_once(APP_INC_PATH . "class.user.php");
 include_once(APP_PEAR_PATH . 'Net/SmartIRC.php');
 
+ini_set("memory_limit", "256M");
+
 // if requested, clear the lock
 if (in_array('--fix-lock', @$HTTP_SERVER_VARS['argv'])) {
     Lock::release('irc_bot');
