diff -u ./modifier.highlight_quoted.php /tmp/modifier.highlight_quoted.php
--- ./modifier.highlight_quoted.php	2005-08-24 11:27:20.000000000 +0300
+++ /tmp/modifier.highlight_quoted.php	2005-08-24 11:26:33.000000000 +0300
@@ -56,18 +56,19 @@
         if ($escape) {
             $match[2]    =    htmlspecialchars($match[2]);
         }
-        if(trim($match[2]) != "") {
+		$line = trim($match[2]);
+		if ($line) {
             $indent =    strlen(preg_replace('/[\s]*/','',$match[1]));
             $color  =    $indent % count($colors);
 
             $ret   .=    '<font color="'.$colors[$color].'">';
             $ret   .=    htmlspecialchars($match[1])." ";
-            $ret   .=    $match[2];
-            $ret   .=    "</font>";
+            $ret   .=    $line;
+            $ret   .=    "</font>\n";
         } else {
-            $ret   .=    $match[2];
+            $ret   .=   "\n";
         }
     }
     return $ret;
 }
-?>
\ No newline at end of file
+?>
