--- eventum-1.4/setup/config.inc.php~	2005-01-25 23:50:53.000000000 +0200
+++ eventum-1.4/setup/config.inc.php	2005-01-25 23:51:23.000000000 +0200
@@ -54,9 +54,9 @@
 }
 @define("APP_PATH", $app_path);
 @define("APP_INC_PATH", APP_PATH . "include/");
-@define("APP_PEAR_PATH", APP_INC_PATH . "pear/");
+@define("APP_PEAR_PATH", "/usr/share/pear/");
 @define("APP_TPL_PATH", APP_PATH . "templates/");
-@define("APP_SMARTY_PATH", APP_INC_PATH . "Smarty/");
+@define("APP_SMARTY_PATH", APP_PEAR_PATH . "Smarty/");
 @define("APP_JPGRAPH_PATH", APP_INC_PATH . "jpgraph/");
 @define("APP_LOG_PATH", "/var/log/eventum/");
 @define("APP_LOCKS_PATH", "/var/run/eventum/");
--- ./rpc/xmlrpc.php~	2005-01-05 01:06:19.000000000 +0200
+++ ./rpc/xmlrpc.php	2005-01-25 16:35:14.000000000 +0200
@@ -41,7 +41,7 @@
 require_once(APP_INC_PATH . "class.template.php");
 require_once(APP_INC_PATH . "class.customer.php");
 error_reporting(0);
-require_once(APP_PEAR_PATH . "XML_RPC/Server.php");
+require_once(APP_PEAR_PATH . "XML/RPC/Server.php");
 
 function authenticate($email, $password)
 {
--- ./include/class.notification.php~	2005-01-05 01:06:19.000000000 +0200
+++ ./include/class.notification.php	2005-01-25 16:41:23.000000000 +0200
@@ -613,9 +613,9 @@
         }
         if ($old["iss_description"] != $new["description"]) {
             // need real diff engine here
-            require_once 'Text_Diff/Diff.php';
-            require_once 'Text_Diff/Diff/Renderer.php';
-            require_once 'Text_Diff/Diff/Renderer/unified.php';
+            require_once 'Text/Diff.php';
+            require_once 'Text/Diff/Renderer.php';
+            require_once 'Text/Diff/Renderer/unified.php';
             $old['iss_description'] = explode("\n", $old['iss_description']);
             $new['description'] = explode("\n", $new['description']);
             $diff = &new Text_Diff($old["iss_description"], $new["description"]);
