--- ../BUILD/eventum-1.4/misc/irc/bot.php	2005-01-05 01:06:16.000000000 +0200
+++ a/misc/irc/bot.php	2005-01-18 16:09:13.000000000 +0200
@@ -52,10 +52,12 @@
     exit;
 }
 
+include '/etc/eventum/irc.php';
+
 // SETUP: need to change the project name in here
 $channels = array(
     Project::getID('Default Project') => array(
-        '#issues',
+        $irc_default_channel,
     )
 );
 
@@ -355,9 +357,8 @@
 $irc->registerActionhandler(SMARTIRC_TYPE_QUERY, '^!?list-clocked-in', $bot, 'listClockedInUsers');
 $irc->registerActionhandler(SMARTIRC_TYPE_QUERY, '^!?list-quarantined', $bot, 'listQuarantinedIssues');
 
-
-$irc->connect('localhost', 6667);
-$irc->login('EventumBOT', 'EventumBOT', 0, 'EventumBOT');
+$irc->connect($irc_host, $irc_port);
+$irc->login($irc_nick, $irc_realname, 0, $irc_username);
 foreach ($channels as $prj_id => $channel_list) {
     $irc->join($channel_list);
 }
@@ -366,4 +367,4 @@
 
 // release the lock
 Lock::release('irc_bot');
-?>
\ No newline at end of file
+?>
