--- eventum-3.7.4-440-gcc5ad3db5/src/Config/Paths.php	2019-08-21 23:32:40.826463576 +0300
+++ eventum-3.7.4-440-gcc5ad3db5/src/Config/Paths.php	2019-08-21 23:38:04.548486455 +0300
@@ -24,14 +24,13 @@
     // "/var" path for writable data
-    public const APP_VAR_PATH = APP_PATH . '/var';
-    public const APP_CACHE_PATH = self::APP_VAR_PATH . '/cache';
+    public const APP_CACHE_PATH = '/var/cache/eventum';
 
     // define other paths
     public const APP_TPL_PATH = APP_PATH . '/templates';
     public const APP_TPL_COMPILE_PATH = self::APP_CACHE_PATH . '/smarty';
-    public const APP_LOG_PATH = self::APP_VAR_PATH . '/log';
+    public const APP_LOG_PATH = '/var/log/eventum';
     public const APP_ERROR_LOG = self::APP_LOG_PATH . '/errors.log';
-    public const APP_LOCKS_PATH = self::APP_VAR_PATH . '/lock';
+    public const APP_LOCKS_PATH = '/var/run/eventum';
 
     // fonts directory for phplot
-    public const APP_FONTS_PATH = APP_PATH . '/vendor/fonts/liberation';
+    public const APP_FONTS_PATH = '/usr/share/fonts/TTF';
 }
--- eventum-3.5.6-98-g5b87d2b0/src/Setup/Requirements.php~	2019-01-23 01:50:42.000000000 +0200
+++ eventum-3.5.6-98-g5b87d2b0/src/Setup/Requirements.php	2019-01-24 22:18:56.047816296 +0200
@@ -64,10 +64,6 @@
         $configPath = Setup::getConfigPath();
         $setupFile = Setup::getSetupFile();
 
-        $error = $this->checkPermissions($configPath, "Directory '" . $configPath . "'", true);
-        if (!empty($error)) {
-            $errors[] = $error;
-        }
         $error = $this->checkPermissions($setupFile, "File '" . $setupFile . "'");
         if (!empty($error)) {
             $errors[] = $error;
--- eventum-3.5.0/src/Console/Command/MonitorCommand.php~	2018-05-20 17:33:34.000000000 +0300
+++ eventum-3.5.0/src/Console/Command/MonitorCommand.php	2018-05-21 01:42:21.463354763 +0300
@@ -47,17 +47,17 @@
         $required_files = [
             $configPath . '/config.php' => [
                 'check_owner' => true,
-                'owner' => 'apache',
+                'owner' => 'root',
                 'check_group' => true,
-                'group' => 'apache',
+                'group' => 'http',
                 'check_permission' => true,
                 'permission' => 640,
             ],
             $configPath . '/setup.php' => [
                 'check_owner' => true,
-                'owner' => 'apache',
+                'owner' => 'root',
                 'check_group' => true,
-                'group' => 'apache',
+                'group' => 'http',
                 'check_permission' => true,
                 'permission' => 660,
                 'check_filesize' => true,
--- eventum-3.1.5/config/sphinx.conf.php~	2016-11-13 12:05:37.000000000 +0200
+++ eventum-3.1.5/config/sphinx.conf.php	2016-11-23 03:10:05.855040510 +0200
@@ -1,7 +1,7 @@
 #!/usr/bin/php
 <?php
 
-require_once __DIR__ . '/../init.php';
+require_once '/usr/share/eventum/init.php';
 
 if (!defined('SPHINX_LOG_PATH')) {
     define('SPHINX_LOG_PATH', '/var/log/sphinx/');
--- eventum-3.7.4-440-gcc5ad3db5/src/Attachment/StorageManager.php~	2019-08-11 22:49:07.000000000 +0300
+++ eventum-3.7.4-440-gcc5ad3db5/src/Attachment/StorageManager.php	2019-08-21 23:33:19.325911295 +0300
@@ -33,7 +33,7 @@
      *
      * @var string
      */
-    private const STORAGE_PATH = Paths::APP_PATH . '/var/storage/';
+    private const STORAGE_PATH = '/var/lib/eventum/storage/';
 
     /**
      * @var MountManager
--- eventum/res/services.yml~	2019-08-02 01:33:18.681001511 +0300
+++ eventum/res/services.yml	2019-08-02 01:33:20.290978222 +0300
@@ -15,13 +15,13 @@
     # makes classes in src/ available to be used as services
     # this creates a service per class whose id is the fully-qualified class name
     Eventum\:
-        resource: '../src/*'
-        exclude: '../src/{DependencyInjection,Entity,Migrations,Tests,Kernel.php}'
+        resource: '/usr/share/eventum/src/*'
+        exclude: '/usr/share/eventum/src/{DependencyInjection,Entity,Migrations,Tests,Kernel.php}'
 
     # controllers are imported separately to make sure services can be injected
     # as action arguments even if you don't extend any base controller class
     Eventum\Controller\:
-        resource: '../src/Controller'
+        resource: '/usr/share/eventum/src/Controller'
         tags: ['controller.service_arguments']
 
     # add more service definitions when explicit configuration is needed
