--- eventum-1.6.0/setup/index.php~	2005-07-30 17:31:20.000000000 +0300
+++ eventum-1.6.0/setup/index.php	2005-07-30 17:45:48.000000000 +0300
@@ -329,7 +329,7 @@
             $user_list = array_map('strtolower', $user_list);
             if (@$HTTP_POST_VARS["create_user"] == 'yes') {
                 if (!in_array(strtolower(@$HTTP_POST_VARS['eventum_user']), $user_list)) {
-                    $stmt = "GRANT SELECT, UPDATE, DELETE, INSERT, ALTER, DROP, CREATE ON " . $HTTP_POST_VARS['db_name'] . ".* TO '" . $HTTP_POST_VARS["eventum_user"] . "'@'%' IDENTIFIED BY '" . $HTTP_POST_VARS["eventum_password"] . "'";
+                    $stmt = "GRANT SELECT, UPDATE, DELETE, INSERT, ALTER, DROP, CREATE, INDEX ON " . $HTTP_POST_VARS['db_name'] . ".* TO '" . $HTTP_POST_VARS["eventum_user"] . "'@'%' IDENTIFIED BY '" . $HTTP_POST_VARS["eventum_password"] . "'";
                     if (!mysql_query($stmt, $conn)) {
                         return getErrorMessage('create_user', mysql_error());
                     }
