--- include/class.support.php	2005-02-18 14:28:18.000000000 +0200
+++ ./include/class.support.php	2005-02-21 17:50:35.000000000 +0200
@@ -556,7 +562,7 @@
             if (($info['ema_issue_auto_creation'] == 'enabled') && ($should_create_issue)) {
                 $options = Email_Account::getIssueAutoCreationOptions($info['ema_id']);
                 $new_issue_id = @Issue::createFromEmail($info['ema_prj_id'], APP_SYSTEM_USER_ID, 
-                        $email->fromaddress, $email->subject, $message_body, $options['category'], 
+                        $email->fromaddress, Mime_Helper::fixEncoding($email->subject), $message_body, $options['category'], 
                         $options['priority'], @$options['users'], $t['date']);
                 $t['issue_id'] = $new_issue_id;
                 // associate any existing replied-to email with this new issue
