--- ./misc/upgrade/v1.4_to_1.5/database_changes.php~	2005-03-03 19:02:11.000000000 +0200
+++ ./misc/upgrade/v1.4_to_1.5/database_changes.php	2005-03-03 19:37:50.000000000 +0200
@@ -5,22 +5,6 @@
 
 $stmts = array();
 
-// see if this category already exists and if not, add it
-$sql = "SELECT
-            count(*)
-        FROM
-            " . APP_DEFAULT_DB . "." . APP_TABLE_PREFIX . "time_tracking_category
-        WHERE
-            ttc_title = 'Email Discussion'";
-$res = $GLOBALS["db_api"]->dbh->getOne($sql);
-if ($res == 0) {
-    // test if this works
-    $stmts[] = "INSERT INTO eventum_time_tracking_category (ttc_id, ttc_title, ttc_created_date) VALUES (9, 'Email Discussion', NOW())";
-}
-
-$stmts[] = "ALTER TABLE eventum_project_user ADD COLUMN pru_role tinyint(1) unsigned default 1";
-$stmts[] = "ALTER TABLE eventum_project ADD COLUMN prj_segregate_reporter tinyint(1) DEFAULT 0";
-$stmts[] = "ALTER TABLE eventum_issue ADD COLUMN iss_private tinyint(1) NOT NULL DEFAULT 0";
 $stmts[] = "ALTER TABLE eventum_email_draft ADD COLUMN emd_status enum('pending', 'edited', 'sent') NOT NULL DEFAULT 'pending' AFTER emd_sup_id";
 $stmts[] = "INSERT INTO eventum_history_type (htt_id, htt_name, htt_role) VALUES (NULL, 'scm_checkin_associated', 0)";
 $stmts[] = "ALTER TABLE eventum_project_priority ADD COLUMN pri_rank TINYINT(1) NOT NULL";
@@ -36,4 +20,4 @@
 }
 
 ?>
-done
\ No newline at end of file
+done
--- ./misc/upgrade/v1.4_to_1.5/database_changes.php~	2005-03-03 19:39:10.000000000 +0200
+++ ./misc/upgrade/v1.4_to_1.5/database_changes.php	2005-03-03 19:40:18.000000000 +0200
@@ -9,6 +9,8 @@
 $stmts[] = "INSERT INTO eventum_history_type (htt_id, htt_name, htt_role) VALUES (NULL, 'scm_checkin_associated', 0)";
 $stmts[] = "ALTER TABLE eventum_project_priority ADD COLUMN pri_rank TINYINT(1) NOT NULL";
 $stmts[] = "UPDATE eventum_columns_to_display SET ctd_field='pri_rank' WHERE ctd_field='iss_pri_id'";
+$stmts[] = "ALTER TABLE eventum_mail_queue ADD COLUMN maq_usr_id int(11) unsigned";
+$stmts[] = "ALTER TABLE eventum_mail_queue ADD COLUMN maq_type varchar(30) DEFAULT ''";
 
 foreach ($stmts as $stmt) {
     $stmt = str_replace('eventum_', APP_TABLE_PREFIX, $stmt);
--- ./misc/upgrade/v1.4_to_1.5/database_changes.php~	2005-03-03 19:41:13.000000000 +0200
+++ ./misc/upgrade/v1.4_to_1.5/database_changes.php	2005-03-03 19:47:08.000000000 +0200
@@ -5,7 +5,6 @@
 
 $stmts = array();
 
-$stmts[] = "ALTER TABLE eventum_email_draft ADD COLUMN emd_status enum('pending', 'edited', 'sent') NOT NULL DEFAULT 'pending' AFTER emd_sup_id";
 $stmts[] = "INSERT INTO eventum_history_type (htt_id, htt_name, htt_role) VALUES (NULL, 'scm_checkin_associated', 0)";
 $stmts[] = "ALTER TABLE eventum_project_priority ADD COLUMN pri_rank TINYINT(1) NOT NULL";
 $stmts[] = "UPDATE eventum_columns_to_display SET ctd_field='pri_rank' WHERE ctd_field='iss_pri_id'";
--- ../misc/upgrade/v1.4_to_1.5/database_changes.php~	2005-03-03 19:54:11.000000000 +0200
+++ ../misc/upgrade/v1.4_to_1.5/database_changes.php	2005-03-03 19:56:28.000000000 +0200
@@ -5,7 +5,6 @@
 
 $stmts = array();
 
-$stmts[] = "INSERT INTO eventum_history_type (htt_id, htt_name, htt_role) VALUES (NULL, 'scm_checkin_associated', 0)";
 $stmts[] = "ALTER TABLE eventum_project_priority ADD COLUMN pri_rank TINYINT(1) NOT NULL";
 $stmts[] = "UPDATE eventum_columns_to_display SET ctd_field='pri_rank' WHERE ctd_field='iss_pri_id'";
 $stmts[] = "ALTER TABLE eventum_mail_queue ADD COLUMN maq_usr_id int(11) unsigned";
