--- ./misc/cli/eventum~	2005-01-19 02:47:20.000000000 +0200
+++ ./misc/cli/eventum	2005-01-19 03:02:40.119100405 +0200
@@ -11,7 +11,7 @@
 //
 // @(#) $Id$
 //
-include_once("config.inc.php");
+include_once("/etc/eventum/cli.php");
 include_once(APP_INC_PATH . "class.command_line.php");
 include_once(APP_PEAR_PATH . "XML_RPC/RPC.php");
 
--- eventum-1.4.orig/misc/cli/config.inc.php	2005-01-26 00:00:58.000000000 +0200
+++ eventum-1.4/misc/cli/config.inc.php	2005-01-25 23:57:39.000000000 +0200
@@ -45,16 +45,10 @@
 }
 
 // definitions of path related variables
-@define("APP_PATH", dirname(__FILE__) . '/');
+@define("APP_PATH", '/usr/share/eventum/cli/');
 @define("APP_INC_PATH", APP_PATH . "include/");
 @define("APP_PEAR_PATH", APP_INC_PATH . "pear/");
-if (stristr(PHP_OS, 'darwin')) {
-    ini_set("include_path", ".:" . APP_PEAR_PATH);
-} elseif (stristr(PHP_OS, 'win')) {
-    ini_set("include_path", ".;" . APP_PEAR_PATH);
-} else {
-    ini_set("include_path", ".:" . APP_PEAR_PATH);
-}
+ini_set("include_path", ".:" . APP_PEAR_PATH);
 
 @define("APP_BENCHMARK", false);
 ?>
--- eventum-1.4/misc/cli/config.inc.php~	2005-02-22 21:35:52.000000000 +0200
+++ eventum-1.4/misc/cli/config.inc.php	2005-02-22 21:44:10.000000000 +0200
@@ -46,9 +46,9 @@
 
 // definitions of path related variables
 @define("APP_PATH", '/usr/share/eventum/cli/');
-@define("APP_INC_PATH", APP_PATH . "include/");
+@define("APP_INC_PATH", APP_PATH);
 @define("APP_PEAR_PATH", APP_INC_PATH . "pear/");
 ini_set("include_path", ".:" . APP_PEAR_PATH);
 
 @define("APP_BENCHMARK", false);
-?>
\ No newline at end of file
+?>
--- ./misc/cli/config.inc.php~	2005-02-22 21:49:37.000000000 +0200
+++ ./misc/cli/config.inc.php	2005-02-22 21:50:01.000000000 +0200
@@ -47,7 +47,7 @@
 // definitions of path related variables
 @define("APP_PATH", '/usr/share/eventum/cli/');
 @define("APP_INC_PATH", APP_PATH);
-@define("APP_PEAR_PATH", APP_INC_PATH . "pear/");
+@define("APP_PEAR_PATH", "/usr/share/pear/");
 ini_set("include_path", ".:" . APP_PEAR_PATH);
 
 @define("APP_BENCHMARK", false);
--- ./misc/cli/eventum~	2005-01-24 17:49:25.000000000 +0200
+++ ./misc/cli/eventum	2005-01-25 15:48:56.000000000 +0200
@@ -13,7 +13,7 @@
 //
 include_once("/etc/webapps/eventum/cli.php");
 include_once(APP_INC_PATH . "class.command_line.php");
-include_once(APP_PEAR_PATH . "XML_RPC/RPC.php");
+include_once(APP_PEAR_PATH . "XML/RPC.php");
 
 list($user_email, $user_password, $url, $port, $relative_url) = Command_Line::getEnvironmentSettings();
 if (empty($port)) {
--- ./misc/cli/include/class.command_line.php	2005-01-05 01:06:03.000000000 +0200
+++ ./misc/cli/include/class.command_line.php	2005-01-25 15:48:56.000000000 +0200
@@ -29,7 +29,7 @@
 //
 
 include_once(APP_INC_PATH . "class.misc.php");
-include_once(APP_PEAR_PATH . "XML_RPC/RPC.php");
+include_once(APP_PEAR_PATH . "XML/RPC.php");
 
 $_displayed_confirmation = false;
 
--- eventum-localization-dev/include/class.template.php~	2006-04-10 23:48:12.740151749 +0300
+++ eventum-localization-dev/include/class.template.php	2006-04-10 23:48:16.310231300 +0300
@@ -60,7 +60,7 @@
     {
         $this->smarty = new Smarty;
         $this->smarty->template_dir = APP_TPL_PATH;
-        $this->smarty->compile_dir = APP_PATH . "templates_c";
+        $this->smarty->compile_dir = '/var/cache/eventum';
         $this->smarty->config_dir = '';
         $this->smarty->register_modifier("activateLinks", array('Link_Filter', 'activateLinks'));
         $this->smarty->register_modifier("formatCustomValue", array('Custom_Field', 'formatValue'));
--- ./include/class.monitor.php~	2005-02-22 21:50:49.000000000 +0200
+++ ./include/class.monitor.php	2005-02-22 23:04:50.000000000 +0200
@@ -157,6 +165,7 @@
                 'permission'       => 100,
             ),
         );
+        $required_directories = array();
         foreach ($required_directories as $dir_path => $options) {
             // check if directory exists
             if (!file_exists($dir_path)) {
--- eventum-20060717/setup/index.php~	2006-07-18 00:20:57.530267420 +0300
+++ eventum-20060717/setup/index.php	2006-07-18 00:22:06.301805247 +0300
@@ -121,31 +121,31 @@
     if (ini_get('allow_call_time_pass_reference') != "1") {
         $errors[] = "The 'allow_call_time_pass_reference' directive needs to be enabled in your PHP.INI file in order for Eventum to work properly.";
     }
-    $error = checkPermissions('../locks', "Directory 'locks'", TRUE);
+    $error = checkPermissions('/var/run/eventum', "Directory '/var/run/eventum'", TRUE);
     if (!empty($error)) {
         $errors[] = $error;
     }
-    $error = checkPermissions('../logs', "Directory 'logs'", TRUE);
+    $error = checkPermissions('/var/log/eventum', "Directory '/var/log/eventum'", TRUE);
     if (!empty($error)) {
         $errors[] = $error;
     }
-    $error = checkPermissions('../templates_c', "Directory 'templates_c'", TRUE);
+    $error = checkPermissions('/var/cache/eventum', "Directory '/var/cache/eventum'", TRUE);
     if (!empty($error)) {
         $errors[] = $error;
     }
-    $error = checkPermissions('../config.inc.php', "File 'config.inc.php'");
+    $error = checkPermissions('/etc/webapps/eventum/config.php', "File '/etc/webapps/eventum/config.php'");
     if (!empty($error)) {
         $errors[] = $error;
     }
-    $error = checkPermissions('../logs/errors.log', "File 'logs/errors.log'");
+    $error = checkPermissions('/var/log/eventum/errors.log', "File '/var/log/eventum/errors.log'");
     if (!empty($error)) {
         $errors[] = $error;
     }
-    $error = checkPermissions('../setup.conf.php', "File 'setup.conf.php'");
+    $error = checkPermissions('/etc/webapps/eventum/setup.php', "File '/etc/webapps/eventum/setup.php'");
     if (!empty($error)) {
         $errors[] = $error;
     }
-    $error = checkPermissions('../include/private_key.php', "File 'include/private_key.php'");
+    $error = checkPermissions('/etc/webapps/eventum/private_key.php', "File '/etc/webapps/eventum/private_key.php'");
     if (!empty($error)) {
         $errors[] = $error;
     }
--- eventum-20060918/setup/index.php~	2006-09-19 00:15:15.216195060 +0300
+++ eventum-20060918/setup/index.php	2006-09-19 00:15:18.346265301 +0300
@@ -276,8 +276,8 @@
 
     clearstatcache();
     // check if config.inc.php in the root directory is writable
-    if (!is_writable('../config.inc.php')) {
-        return "The file 'config.inc.php' in Eventum's root directory needs to be writable by the web server user. Please correct this problem and try again.";
+    if (!is_writable('/etc/webapps/eventum/config.php')) {
+        return "The file '/etc/webapps/eventum/config.php' needs to be writable by the web server user. Please correct this problem and try again.";
     }
     // gotta check and see if the provided installation path really exists...
     if (!file_exists($_POST['path'])) {
@@ -287,16 +287,16 @@
     $private_key = '<?php
 $private_key = "' . md5(microtime()) . '";
 ?>';
-    if (!is_writable('../include/private_key.php')) {
-        return "The file 'include/private_key.php' needs to be writable by the web server user. Please correct this problem and try again.";
+    if (!is_writable('/etc/webapps/eventum/private_key.php')) {
+        return "The file '/etc/webapps/eventum/private_key.php' needs to be writable by the web server user. Please correct this problem and try again.";
     }
-    $fp = @fopen('../include/private_key.php', 'w');
+    $fp = @fopen('/etc/webapps/eventum/private_key.php', 'w');
     if ($fp === FALSE) {
-        return "Could not open the file 'include/private_key.php' for writing. The permissions on the file should be set as to allow the user that the web server runs as to open it. Please correct this problem and try again.";
+        return "Could not open the file '/etc/webapps/eventum/private_key.php' for writing. The permissions on the file should be set as to allow the user that the web server runs as to open it. Please correct this problem and try again.";
     }
     $res = fwrite($fp, $private_key);
     if ($fp === FALSE) {
-        return "Could not write the configuration information to 'include/private_key.php'. The file should be writable by the user that the web server runs as. Please correct this problem and try again.";
+        return "Could not write the configuration information to '/etc/webapps/eventum/private_key.php'. The file should be writable by the user that the web server runs as. Please correct this problem and try again.";
     }
     fclose($fp);
     // check if we can connect
@@ -429,18 +429,7 @@
     $tpl->assign('is_imap_enabled', function_exists('imap_open'));
 }
 
-
-$full_url = dirname($_SERVER['PHP_SELF']);
-$pieces = explode("/", $full_url);
-$relative_url = array();
-$relative_url[] = '';
-foreach ($pieces as $piece) {
-    if ((!empty($piece)) && ($piece != 'setup')) {
-        $relative_url[] = $piece;
-    }
-}
-$relative_url[] = '';
-$relative_url = implode("/", $relative_url);
+$relative_url = '/eventum/';
 
 $tpl->assign("phpversion", phpversion());
 $tpl->assign("rel_url", $relative_url);
--- ./index.php~	2005-04-22 18:33:06.556398818 +0300
+++ ./index.php	2005-04-22 18:33:07.885180913 +0300
@@ -33,8 +33,8 @@
 include_once(APP_INC_PATH . "db_access.php");
 
 // check if templates_c is writable by the web server user
-if (!Misc::isWritableDirectory(APP_PATH . 'templates_c')) {
-    $errors = array("Directory 'templates_c' is not writable.");
+if (!Misc::isWritableDirectory('/var/cache/eventum')) {
+    $errors = array("Directory '/var/cache/eventum' is not writable.");
     Misc::displayRequirementErrors($errors);
     exit;
 }
--- ../include/class.support.php~	2005-07-10 15:40:57.000000000 +0300
+++ ../include/class.support.php	2005-07-14 22:47:58.665843734 +0300
@@ -259,7 +259,7 @@
      */
     function saveRoutedEmail($message)
     {
-        $path = APP_PATH . "misc/routed_emails/";
+        $path = "/var/lib/eventum/routed_emails/";
         list($usec,) = explode(" ", microtime());
         $filename = date('dmY.His.') . $usec . '.email.txt';
         $fp = @fopen($path . $filename, 'w');
--- eventum-20051221/include/class.draft.php~	2005-12-21 17:56:27.000000000 +0200
+++ eventum-20051221/include/class.draft.php	2005-12-21 17:59:26.000000000 +0200
@@ -40,7 +40,7 @@
      */
     function saveRoutedMessage($message)
     {
-        $path = APP_PATH . "misc/routed_drafts/";
+        $path = "/var/lib/eventum/routed_drafts/";
         list($usec,) = explode(" ", microtime());
         $filename = date('dmY.His.') . $usec . '.draft.txt';
         $fp = @fopen($path . $filename, 'w');
--- eventum-20051221/include/class.note.php~	2005-12-21 17:56:27.000000000 +0200
+++ eventum-20051221/include/class.note.php	2005-12-21 17:59:57.000000000 +0200
@@ -255,7 +255,7 @@
      */
     function saveRoutedNote($message)
     {
-        $path = APP_PATH . "misc/routed_notes/";
+        $path = "/var/lib/eventum/routed_notes/";
         list($usec,) = explode(" ", microtime());
         $filename = date('dmY.His.') . $usec . '.note.txt';
         $fp = @fopen($path . $filename, 'w');
--- ../setup/index.php~	2005-07-26 18:49:37.000000000 +0300
+++ ../setup/index.php	2005-07-26 19:12:57.000000000 +0300
@@ -417,7 +417,7 @@
         $config_contents = str_replace("%{APP_ENABLE_FULLTEXT}%", "false", $config_contents);
     }
 
-    $fp = @fopen('../config.inc.php', 'w');
+    $fp = @fopen('/etc/webapps/eventum/config.php', 'w');
     if ($fp === FALSE) {
         return "Could not open the file 'config.inc.php' for writing. The permissions on the file should be set as to allow the user that the web server runs as to open it. Please correct this problem and try again.";
     }
--- eventum-localization-dev2/setup/index.php~	2006-06-02 00:58:11.321514674 +0300
+++ eventum-localization-dev2/setup/index.php	2006-06-02 00:58:46.032294537 +0300
@@ -208,11 +208,11 @@
     exit;
 }
 
-include_once("../include/Smarty/Smarty.class.php");
+include_once("/usr/share/pear/Smarty/Smarty.class.php");
 
 $tpl = new Smarty();
-$tpl->template_dir = '../templates/';
-$tpl->compile_dir = "../templates_c";
+$tpl->template_dir = '/usr/share/eventum/templates';
+$tpl->compile_dir = "/var/cache/eventum";
 $tpl->config_dir = '';
 
 function replace_table_prefix($str)
--- eventum-1.5.5/setup/index.php~	2005-07-26 20:22:06.000000000 +0300
+++ eventum-1.5.5/setup/index.php	2005-07-26 20:35:31.000000000 +0300
@@ -198,13 +198,7 @@
     return $html;
 }
 
-if (stristr(PHP_OS, 'darwin')) {
-    ini_set("include_path", ".:./../include/pear/");
-} elseif (stristr(PHP_OS, 'win')) {
-    ini_set("include_path", ".;./../include/pear/");
-} else {
-    ini_set("include_path", ".:./../include/pear/");
-}
+ini_set("include_path", ".:../../include/pear/");
 include_once("File/Util.php");
 
 $html = checkRequirements();
--- eventum-localization-dev/include/class.template.php~	2006-04-11 00:16:59.258616656 +0300
+++ eventum-localization-dev/include/class.template.php	2006-04-11 00:17:23.649160189 +0300
@@ -61,6 +61,7 @@
         $this->smarty = new Smarty;
         $this->smarty->template_dir = APP_TPL_PATH;
         $this->smarty->compile_dir = '/var/cache/eventum';
+        $this->smarty->plugins_dir  = array('/usr/share/eventum/include/smarty', 'plugins');
         $this->smarty->config_dir = '';
         $this->smarty->register_modifier("activateLinks", array('Link_Filter', 'activateLinks'));
         $this->smarty->register_modifier("formatCustomValue", array('Custom_Field', 'formatValue'));
--- ./include/class.misc.php	2005-10-26 17:17:17.000000000 +0300
+++ /tmp/class.misc.php	2005-10-26 17:30:12.000000000 +0300
@@ -607,7 +607,7 @@
      */
     function highlightQuotedReply($text)
     {
-        include_once(APP_SMARTY_PATH . "plugins/modifier.highlight_quoted.php");
+        include_once(APP_INC_PATH. "smarty/modifier.highlight_quoted.php");
         return smarty_modifier_highlight_quoted($text);
     }
 
