--- ../../eventum-1.5.2.orig/reports/recent_activity.php	2005-04-16 05:53:27.000000000 +0300
+++ ./reports/recent_activity.php	2005-04-16 06:02:20.000000000 +0300
@@ -297,10 +297,18 @@
             $res[$i]["customer"] = @$details['customer_name'];
         }
         $res[$i]["date"] = Date_API::getFormattedDate($res[$i][$date_field], Date_API::getPreferredTimezone($usr_id));
+        # need to decode From:, To: mail headers
+        if ($res[$i]["sup_from"]) {
+            $res[$i]["sup_from"] = Mime_Helper::fixEncoding($res[$i]["sup_from"]);
+        }
+        if ($res[$i]["sup_to"]) {
+            $res[$i]["sup_to"] = Mime_Helper::fixEncoding($res[$i]["sup_to"]);
+        }
+
         $data[] = $res[$i];
     }
     return $data;
 }
 
 $tpl->displayTemplate();
-?>
\ No newline at end of file
+?>
