--- eventum-20060830/include/class.notification.php	2006-08-30 20:17:25.074832296 +0300
+++ eventum/include/class.notification.php	2006-09-21 20:59:32.125474606 +0300
@@ -853,6 +853,30 @@
         }
     }
 
+    /**
+     * Method used to get list of addresses that were email sent to.
+     *
+     * @access  public
+     * @param   integer $issue_id The issue ID
+     * @return  array   list of addresse
+     */
+    function getLastNotifiedAddresses($issue_id = null)
+    {
+        global $_EVENTUM_LAST_NOTIFIED_LIST;
+
+        if (is_null($_EVENTUM_LAST_NOTIFIED_LIST)) {
+            return null;
+        }
+
+        if (is_null($issue_id)) {
+            // return all addresses in flat view
+            return array_values($_EVENTUM_LAST_NOTIFIED_LIST);
+        }
+
+        // return address list for specific issue_id only.
+        return $_EVENTUM_LAST_NOTIFIED_LIST[$issue_id];
+    }
+
 
     /**
      * Method used to format and send the email notifications.
@@ -869,6 +893,8 @@
      */
     function notifySubscribers($issue_id, $emails, $type, $data, $subject, $internal_only, $type_id = false, $headers = false)
     {
+        global $_EVENTUM_LAST_NOTIFIED_LIST;
+
         // open text template
         $tpl = new Template_API;
         $tpl->setTemplate('notifications/' . $type . '.tpl.text');
@@ -954,6 +980,8 @@
                 $full_subject = "[#$issue_id] $subject: $extra_subject";
             }
             $mail->send($from, $emails[$i], $full_subject, TRUE, $issue_id, $final_type, $sender_usr_id, $type_id);
+
+            $_EVENTUM_LAST_NOTIFIED_LIST[$issue_id][] = $emails[$i];
         }
     }
 
--- eventum-20060829/update.php	2006-08-30 00:46:19.047416365 +0300
+++ eventum-20060830/update.php	2006-08-30 20:17:24.844826982 +0300
@@ -111,6 +111,7 @@
 
     $tpl->assign(array(
         "subscribers"  => Notification::getSubscribers($issue_id),
+        "notify_list"  => Notification::getLastNotifiedAddresses($issue_id),
         "categories"   => Category::getAssocList($prj_id),
         "priorities"   => Priority::getAssocList($prj_id),
         "status"       => $statuses,
--- eventum-20060829/templates/update_form.tpl.html	2006-08-30 00:46:18.777410305 +0300
+++ eventum-20060830/templates/update_form.tpl.html	2006-08-30 20:21:39.160697745 +0300
@@ -10,12 +10,23 @@
             {if $update_result == -1}
               <b>{t}Sorry, an error happened while trying to run your query.{/t}</b>
             {elseif $update_result == 1}
-              <b>{t}Thank you, issue #{$smarty.post.issue_id} was updated successfully.{/t}
+              <b>{t}Thank you, issue #{$smarty.post.issue_id} was updated successfully.{/t}</b>
               {if $has_duplicates == 'yes'}
                 {t}Also, all issues that are marked as duplicates from this one were updated as well.{/t}
               {/if}
               <br /><br />
+
+              {if $notify_list|@count > 0}
+              <b>{t}E-mail about issue update was sent to:{/t}</b><br />
+              {foreach from=$notify_list item=email_address}
+              &nbsp;&nbsp;{$email_address|escape:"html"}<br />
+              {/foreach}
+              <br />
+              {/if}
+
+              <b>
               <a href="view.php?id={$smarty.post.issue_id}" class="link">{t}Return to Issue{/t} #{$smarty.post.issue_id} {t}Details Page{/t}</a>
+              </b>
             {/if}
           </td>
         </tr>
--- eventum-20060830/close.php	2006-09-04 14:46:53.766630207 +0300
+++ eventum-20060830/close.php	2006-09-04 13:57:56.000000000 +0300
@@ -68,9 +68,12 @@
     $tpl->assign("close_result", $res);
 }
 
-$tpl->assign("statuses", Status::getClosedAssocList($prj_id));
-$tpl->assign("resolutions", Resolution::getAssocList());
-$tpl->assign("time_categories", Time_Tracking::getAssocCategories());
+$tpl->assign(array(
+    "statuses" => Status::getClosedAssocList($prj_id),
+    "resolutions" => Resolution::getAssocList(),
+    "time_categories" => Time_Tracking::getAssocCategories(),
+    "notify_list"  => Notification::getLastNotifiedAddresses($issue_id),
+));
 
 if ((Customer::hasCustomerIntegration($prj_id)) && (Customer::hasPerIncidentContract($prj_id, Issue::getCustomerID($issue_id)))) {
     $details = Issue::getDetails($issue_id);
--- eventum-20060830/templates/close.tpl.html~	2006-09-04 14:46:53.486623924 +0300
+++ eventum-20060830/templates/close.tpl.html	2006-09-04 14:45:42.000000000 +0300
@@ -12,7 +12,18 @@
             {if $close_result == -1}
             <b>{t}Sorry, an error happened while trying to run your query.{/t}</b>
             {elseif $close_result == 1}
-            <b>{t}Thank you, the issue was closed successfully. Please choose from one of the options below:{/t}</b>
+            <b>{t}Thank you, the issue was closed successfully.{/t}</b><br/>
+
+            {if $notify_list|@count > 0}
+            <br/>
+            <b>{t}E-mail about issue update was sent to:{/t}</b><br />
+            {foreach from=$notify_list item=email_address}
+            &nbsp;&nbsp;{$email_address|escape:"html"}<br />
+            {/foreach}
+            <br />
+            {/if}
+
+            <b>{t}Please choose from one of the options below:{/t}</b>
             <ul>
               <li><a href="view.php?id={$smarty.post.issue_id}" class="link">{t}Open the Issue Details Page{/t}</a></li>
               <li><a href="list.php" class="link">{t}Open the Issue Listing Page{/t}</a></li>
