--- eventum-3.0.3-287-ge48a790/init.php~	2015-10-26 21:42:14.000000000 +0200
+++ eventum-3.0.3-287-ge48a790/init.php	2015-10-26 21:46:47.983157509 +0200
@@ -76,9 +76,9 @@
 $define('APP_SETUP_FILE', APP_CONFIG_PATH . '/setup.php');
 $define('APP_INC_PATH', APP_PATH . '/lib/eventum');
 $define('APP_TPL_PATH', APP_PATH . '/templates');
-$define('APP_TPL_COMPILE_PATH', APP_VAR_PATH . '/cache');
-$define('APP_LOCKS_PATH', APP_VAR_PATH . '/lock');
-$define('APP_LOG_PATH', APP_VAR_PATH . '/log');
+$define('APP_TPL_COMPILE_PATH', '/var/cache/eventum');
+$define('APP_LOCKS_PATH', '/var/run/eventum');
+$define('APP_LOG_PATH', '/var/log/eventum');
 $define('APP_ERROR_LOG', APP_LOG_PATH . '/errors.log');
 $define('APP_CLI_LOG', APP_LOG_PATH . '/cli.log');
 $define('APP_IRC_LOG', APP_LOG_PATH . '/irc_bot.log');
--- eventum-3.0.3-287-ge48a790/htdocs/setup/index.php~	2015-10-26 21:45:35.367554219 +0200
+++ eventum-3.0.3-287-ge48a790/htdocs/setup/index.php	2015-10-26 21:47:31.775849576 +0200
@@ -47,10 +47,10 @@
 define('APP_CONFIG_PATH', APP_PATH . '/config');
 define('APP_SETUP_FILE', APP_CONFIG_PATH . '/setup.php');
 define('APP_TPL_PATH', APP_PATH . '/templates');
-define('APP_TPL_COMPILE_PATH', APP_VAR_PATH . '/cache');
-define('APP_LOG_PATH', APP_VAR_PATH . '/log');
+define('APP_TPL_COMPILE_PATH', '/var/cache/eventum');
+define('APP_LOG_PATH', '/var/log/eventum');
 define('APP_ERROR_LOG', APP_LOG_PATH . '/errors.log');
-define('APP_LOCKS_PATH', APP_VAR_PATH . '/lock');
+define('APP_LOCKS_PATH', '/var/run/eventum');
 define('APP_LOCAL_PATH', APP_CONFIG_PATH);
 
 header('Content-Type: text/html; charset=' . APP_CHARSET);
@@ -254,10 +254,6 @@
         $errors[] = "The 'file_uploads' directive needs to be enabled in your PHP.INI file in order for Eventum to work properly.";
     }
 
-    $error = checkPermissions(APP_CONFIG_PATH, "Directory '" . APP_CONFIG_PATH . "'", true);
-    if (!empty($error)) {
-        $errors[] = $error;
-    }
     $error = checkPermissions(APP_SETUP_FILE, "File '" . APP_SETUP_FILE. "'");
     if (!empty($error)) {
         $errors[] = $error;
--- eventum-3.0.8-100-g3543f89/src/Command/MonitorCommand.php~	2016-01-31 18:24:19.000000000 +0200
+++ eventum-3.0.8-100-g3543f89/src/Command/MonitorCommand.php	2016-02-01 00:16:28.751423105 +0200
@@ -31,17 +31,17 @@
         $required_files = array(
             APP_CONFIG_PATH . '/config.php' => array(
                 'check_owner' => true,
-                'owner' => 'apache',
+                'owner' => 'root',
                 'check_group' => true,
-                'group' => 'apache',
+                'group' => 'http',
                 'check_permission' => true,
                 'permission' => 640,
             ),
             APP_CONFIG_PATH . '/setup.php' => array(
                 'check_owner' => true,
-                'owner' => 'apache',
+                'owner' => 'root',
                 'check_group' => true,
-                'group' => 'apache',
+                'group' => 'http',
                 'check_permission' => true,
                 'permission' => 660,
                 'check_filesize' => true,
--- eventum-3.0.3-314-gedea8e4/config/sphinx.conf.php~	2015-10-27 18:02:36.000000000 +0200
+++ eventum-3.0.3-314-gedea8e4/config/sphinx.conf.php	2015-10-29 11:53:26.068169385 +0200
@@ -1,7 +1,7 @@
 #!/usr/bin/php
 <?php
 
-require_once __DIR__ . "/../init.php";
+require_once '/usr/share/eventum/init.php';
 
 if (!defined('SPHINX_LOG_PATH')) {
     define('SPHINX_LOG_PATH', '/var/log/sphinx/');
