--- trunk/route_drafts.php~	2009-06-27 17:36:08.000000000 +0300
+++ trunk/route_drafts.php	2009-06-27 17:36:29.036049928 +0300
@@ -29,6 +29,7 @@
 //
 // @(#) $Id$
 
+ini_set('memory_limit', '1024M');
 require_once 'init.php';
 
 $full_message = Misc::getInput();
--- trunk/route_emails.php~	2009-06-27 17:36:08.000000000 +0300
+++ trunk/route_emails.php	2009-06-27 17:36:57.725685544 +0300
@@ -29,6 +29,7 @@
 //
 // @(#) $Id$
 
+ini_set('memory_limit', '1024M');
 require_once 'init.php';
 
 $full_message = Misc::getInput();
--- trunk/route_notes.php~	2009-06-27 17:36:08.000000000 +0300
+++ trunk/route_notes.php	2009-06-27 17:37:14.485553825 +0300
@@ -29,6 +29,7 @@
 //
 // @(#) $Id$
 
+ini_set('memory_limit', '1024M');
 require_once 'init.php';
 
 $full_message = Misc::getInput();
--- eventum-20060830/download_emails.php	2006-09-05 00:11:48.396726556 +0300
+++ eventum-20060830.mem/download_emails.php	2006-09-05 00:12:09.647203129 +0300
@@ -28,7 +28,7 @@
 //
 // @(#) $Id$
 
-ini_set("memory_limit", "256M");
+ini_set("memory_limit", "1024M");
 
 require_once("/etc/webapps/eventum/core.php");
 require_once(APP_INC_PATH . "class.support.php");
--- eventum-20060830/process_mail_queue.php	2006-09-05 00:11:48.396726556 +0300
+++ eventum-20060830.mem/process_mail_queue.php	2006-09-05 00:12:09.647203129 +0300
@@ -28,7 +28,7 @@
 //
 // @(#) $Id$
 
-ini_set("memory_limit", "256M");
+ini_set("memory_limit", '1024M');
 
 require_once("/etc/webapps/eventum/core.php");
 require_once(APP_INC_PATH . "db_access.php");
--- eventum/irc/eventum-irc-bot~	2007-01-16 21:24:17.009987152 +0200
+++ eventum/irc/eventum-irc-bot	2007-01-16 21:24:57.050882112 +0200
@@ -26,6 +26,6 @@
 //
 
-ini_set('memory_limit', '256M');
+ini_set('memory_limit', '1024M');
 
 require_once('/etc/webapps/eventum/core.php');
 require_once('/etc/eventum/irc.php');
