--- eventum-3.10.1/src/Config/Paths.php~	2020-12-02 23:16:35.000000000 +0200
+++ eventum-3.10.1/src/Config/Paths.php	2021-03-24 21:56:18.672527936 +0200
@@ -22,18 +22,16 @@
     public const APP_PUBLIC_PATH = APP_PATH . '/htdocs';
     public const APP_INC_PATH = APP_PATH . '/lib/eventum';
 
-    // "/var" path for writable data
-    private const APP_VAR_PATH = APP_PATH . '/var';
-    public const APP_SPOOL_PATH = self::APP_VAR_PATH . '/spool';
-    public const APP_CACHE_PATH = self::APP_VAR_PATH . '/cache';
+    public const APP_SPOOL_PATH = '/var/spool/eventum';
+    public const APP_CACHE_PATH = '/var/cache/eventum';
 
     // define other paths
     public const APP_TPL_PATH = APP_PATH . '/templates';
     public const APP_TPL_COMPILE_PATH = self::APP_CACHE_PATH . '/smarty';
-    public const APP_LOG_PATH = self::APP_VAR_PATH . '/log';
-    public const APP_LOCKS_PATH = self::APP_VAR_PATH . '/lock';
+    public const APP_LOG_PATH = '/var/log/eventum';
+    public const APP_LOCKS_PATH = '/var/run/eventum';
 
     // fonts directory for phplot
-    public const APP_FONTS_PATH = APP_PATH . '/vendor/fonts/liberation';
+    public const APP_FONTS_PATH = '/usr/share/fonts/TTF';
     public const APP_HELP_PATH = APP_PATH . '/docs/help';
 }
--- eventum-3.9.11.org/src/Setup/Requirements.php	2020-08-01 20:21:08.000000000 +0200
+++ eventum-3.9.11/src/Setup/Requirements.php	2021-03-10 15:18:12.361997906 +0100
@@ -69,10 +69,6 @@ class Requirements
         $privateKeyPath = Setup::getPrivateKeyPath();
         $setupFile = Setup::getSetupFile();
 
-        $error = $this->checkPermissions($configPath, "Directory '" . $configPath . "'", true);
-        if (!empty($error)) {
-            $errors[] = $error;
-        }
         $error = $this->checkPermissions($setupFile, "File '" . $setupFile . "'");
         if (!empty($error)) {
             $errors[] = $error;
--- eventum-3.9.0/src/Console/Command/MonitorCommand.php~	2019-10-21 22:17:43.000000000 +0300
+++ eventum-3.9.0/src/Console/Command/MonitorCommand.php	2020-07-02 00:40:16.042698823 +0300
@@ -68,9 +68,9 @@
         $required_files = [
             $configPath . '/setup.php' => [
                 'check_owner' => true,
-                'owner' => 'apache',
+                'owner' => 'http',
                 'check_group' => true,
-                'group' => 'apache',
+                'group' => 'http',
                 'check_permission' => true,
                 'permission' => 660,
                 'check_filesize' => true,
--- eventum-3.9.11.org/config/sphinx.conf.php	2020-05-05 08:20:33.000000000 +0200
+++ eventum-3.9.11/config/sphinx.conf.php	2021-03-10 15:18:12.361997906 +0100
@@ -4,7 +4,7 @@
 use Eventum\Config\SphinxConfig;
 use Eventum\ServiceContainer;
 
-require_once __DIR__ . '/../init.php';
+require_once '/usr/share/eventum/init.php';
 
 /** @var SphinxConfig $config */
 $config = ServiceContainer::get(SphinxConfig::class);
--- eventum-3.7.4-440-gcc5ad3db5/src/Attachment/StorageManager.php~	2019-08-11 22:49:07.000000000 +0300
+++ eventum-3.7.4-440-gcc5ad3db5/src/Attachment/StorageManager.php	2019-08-21 23:33:19.325911295 +0300
@@ -33,7 +33,7 @@
      *
      * @var string
      */
-    private const STORAGE_PATH = Paths::APP_PATH . '/var/storage/';
+    private const STORAGE_PATH = '/var/lib/eventum/storage/';
 
     /**
      * @var MountManager
