--- eventum-3.0.0-pre1/scm/eventum-cvs-hook.php~	2014-11-30 13:48:18.000000000 +0200
+++ eventum-3.0.0-pre1/scm/eventum-cvs-hook.php	2015-02-09 14:39:22.881991719 +0200
@@ -52,9 +52,7 @@
 // SCM repository name. Needed if multiple repositories configured
 $scm_name = 'cvs';
 
-//
-// DO NOT CHANGE ANYTHING AFTER THIS LINE
-//
+require_once '/etc/eventum/scm.php';
 
 // save name of this script
 $PROGRAM = basename(realpath(array_shift($argv)), '.php');
--- eventum-3.0.0-pre1/scm/eventum-svn-hook.php~	2014-11-30 13:48:18.000000000 +0200
+++ eventum-3.0.0-pre1/scm/eventum-svn-hook.php	2015-02-09 14:40:05.184671966 +0200
@@ -48,9 +48,7 @@
 // SCM repository name. Needed if multiple repositories configured
 $scm_name = 'svn';
 
-//
-// DO NOT CHANGE ANYTHING AFTER THIS LINE
-//
+require_once '/etc/eventum/scm.php';
 
 // save name of this script
 $PROGRAM = basename(realpath(array_shift($argv)), '.php');
