--- eventum/include/class.mime_helper.php~	2006-09-07 23:32:17.000000000 +0300
+++ eventum/include/class.mime_helper.php	2006-09-08 00:17:21.483076433 +0300
@@ -446,12 +446,16 @@
      * Method used to check whether a given email message has any attachments.
      *
      * @access  public
-     * @param   string $message The full body of the message
+     * @param   mixed   $message The full body of the message or parsed message structure.
      * @return  boolean
      */
     function hasAttachments($message)
     {
-        $output = Mime_Helper::decode($message, true);
+        if (is_string($message)) {
+            $output = Mime_Helper::decode($message, true);
+        } else {
+            $output = $message;
+        }
         $attachments = Mime_Helper::_getAttachmentDetails($output, TRUE);
         if (count($attachments) > 0) {
             return true;
--- eventum/include/class.mime_helper.php~	2006-09-08 00:17:21.000000000 +0300
+++ eventum/include/class.mime_helper.php	2006-09-08 00:24:49.529222316 +0300
@@ -57,12 +57,16 @@
      * Method used to get charset from raw email.
      *
      * @access  public
-     * @param   string $input The full body of the message
+     * @param   mixed   $input The full body of the message or decoded email.
      * @return  string charset extracted from Content-Type header of email.
      */
     function getCharacterSet($input)
     {
-        $structure = Mime_Helper::decode($input, false, false);
+        if (!is_object($input)) {
+            $input = Mime_Helper::decode($input, false, false);
+        } else {
+            $structure = $input;
+        }
         if (empty($structure)) {
             return false;
         }
--- eventum/include/class.mime_helper.php~	2006-09-08 00:28:40.000000000 +0300
+++ eventum/include/class.mime_helper.php	2006-09-08 00:52:25.642013349 +0300
@@ -455,12 +455,10 @@
      */
     function hasAttachments($message)
     {
-        if (is_string($message)) {
-            $output = Mime_Helper::decode($message, true);
-        } else {
-            $output = $message;
+        if (!is_object($message)) {
+            $message = Mime_Helper::decode($message, true);
         }
-        $attachments = Mime_Helper::_getAttachmentDetails($output, TRUE);
+        $attachments = Mime_Helper::_getAttachmentDetails($message, TRUE);
         if (count($attachments) > 0) {
             return true;
         } else {
@@ -474,13 +472,15 @@
      * associated with a message.
      *
      * @access  public
-     * @param   string $message The full body of the message
+     * @param   mixed   $message The full body of the message or parsed message structure.
      * @return  array The list of attachments, if any
      */
     function getAttachments($message)
     {
-        $output = Mime_Helper::decode($message, true);
-        return Mime_Helper::_getAttachmentDetails($output, TRUE);
+        if (!is_object($message)) {
+            $message = Mime_Helper::decode($message, true);
+        }
+        return Mime_Helper::_getAttachmentDetails($message, TRUE);
     }
 
 
@@ -489,14 +489,15 @@
      * associated with a message.
      *
      * @access  public
-     * @param   string $message The full body of the message
+     * @param   mixed   $message The full body of the message or parsed message structure.
      * @return  array The list of attachment CIDs, if any
      */
-    function getAttachmentCIDs(&$message)
+    function getAttachmentCIDs($message)
     {
-        // gotta parse MIME based emails now
-        $output = Mime_Helper::decode($message, true);
-        return Mime_Helper::_getAttachmentDetails($output, TRUE);
+        if (!is_object($message)) {
+            $message = Mime_Helper::decode($message, true);
+        }
+        return Mime_Helper::_getAttachmentDetails($message, TRUE);
     }
 
 
@@ -509,6 +510,7 @@
                 $attachments = array_merge($t, $attachments);
             }
         }
+        // FIXME: content-type is always lowered by PEAR class (CHECKME) and why not $mime_part->content_type?
         $content_type = strtolower(@$mime_part->ctype_primary . '/' . @$mime_part->ctype_secondary);
         if ($content_type == '/') {
             $content_type = '';
@@ -567,7 +569,7 @@
      * attachment.
      *
      * @access  public
-     * @param   string $message The full content of the message
+     * @param   mixed   $message The full content of the message or parsed message structure.
      * @param   string $filename The filename to look for
      * @param   string $cid The content-id to look for, if any
      * @return  string The full encoded content of the attachment
@@ -575,8 +577,10 @@
     function getAttachment($message, $filename, $cid = FALSE)
     {
         $parts = array();
-        $output = Mime_Helper::decode($message, true);
-        $details = Mime_Helper::_getAttachmentDetails($output, TRUE, $filename, $cid);
+        if (!is_object($message)) {
+            $message = Mime_Helper::decode($message, true);
+        }
+        $details = Mime_Helper::_getAttachmentDetails($message, TRUE, $filename, $cid);
         if (count($details) == 1) {
             return array(
                 $details[0]['filetype'],
@@ -596,7 +600,7 @@
      * @param   boolean $include_bodies Whether to include the bodies in the return value or not
      * @return  mixed The decoded content of the message
      */
-    function decode(&$message, $include_bodies = FALSE, $decode_bodies = TRUE)
+    function decode($message, $include_bodies = FALSE, $decode_bodies = TRUE)
     {
         // need to fix a pretty annoying bug where if the 'boundary' part of a
         // content-type header is split into another line, the PEAR library would
--- eventum/include/class.support.php	2006-09-08 01:03:12.090024412 +0300
+++ eventum/include/class.support.php	2006-09-08 01:25:16.661067108 +0300
@@ -12,14 +12,17 @@
         }
 
         $email = @imap_headerinfo($mbox, $num);
-        $body = imap_body($mbox, $num);
         $headers = imap_fetchheader($mbox, $num);
-        $message = $headers . $body;
+        $body = imap_body($mbox, $num);
         // check for mysterious blank messages
-        if (empty($message)) {
+        if (empty($body) and empty($headers)) {
             return '';
         }
         $message_id = Mail_API::getMessageID($headers, $body);
+        $message = $headers . $body;
+        // we don't need $body anymore -- free memory
+        unset($body);
+
         if ((!Support::exists($message_id)) && (!Note::exists($message_id))) {
             $structure = Mime_Helper::decode($message, true, true);
             $message_body = Mime_Helper::getMessageBody($structure);
@@ -496,7 +499,7 @@
         if ((!Support::exists($message_id)) && (!Note::exists($message_id))) {
             $structure = Mime_Helper::decode($message, true, true);
             $message_body = Mime_Helper::getMessageBody($structure);
-            if (Mime_Helper::hasAttachments($message)) {
+            if (Mime_Helper::hasAttachments($structure)) {
                 $has_attachments = 1;
             } else {
                 $has_attachments = 0;
@@ -1417,7 +1420,7 @@
             for ($i = 0; $i < count($res); $i++) {
                 // since downloading email should make the emails 'public', send 'false' below as the 'internal_only' flag
                 $structure = Mime_Helper::decode($res[$i]['seb_full_email'], true, false);
-                if (Mime_Helper::hasAttachments($res[$i]['seb_full_email'])) {
+                if (Mime_Helper::hasAttachments($structure)) {
                     $has_attachments = 1;
                 } else {
                     $has_attachments = 0;
--- eventum/include/class.routing.php~	2006-09-08 00:56:22.000000000 +0300
+++ eventum/include/class.routing.php	2006-09-08 01:06:28.235959347 +0300
@@ -169,7 +169,7 @@
             $has_magic_cookie = false;
         }
 
-        if (Mime_Helper::hasAttachments($full_message)) {
+        if (Mime_Helper::hasAttachments($structure)) {
             $has_attachments = 1;
         } else {
             $has_attachments = 0;
@@ -371,7 +371,7 @@
 
         // add the full email to the note if there are any attachments
         // this is needed because the front end code will display attachment links
-        if (Mime_Helper::hasAttachments($full_message)) {
+        if (Mime_Helper::hasAttachments($structure)) {
             $HTTP_POST_VARS['blocked_msg'] = $full_message;
         }
         $res = Note::insert(Auth::getUserID(), $issue_id, false, false);
--- eventum/include/class.note.php~	2006-09-08 00:56:22.000000000 +0300
+++ eventum/include/class.note.php	2006-09-08 01:07:07.482745790 +0300
@@ -524,7 +524,7 @@
         $body = Mime_Helper::getMessageBody($structure);
         $sender_email = strtolower(Mail_API::getEmailAddress($structure->headers['from']));
         if ($target == 'email') {
-            if (Mime_Helper::hasAttachments($blocked_message)) {
+            if (Mime_Helper::hasAttachments($structure)) {
                 $has_attachments = 1;
             } else {
                 $has_attachments = 0;
--- eventum/include/class.mime_helper.php~	2006-09-08 00:56:22.000000000 +0300
+++ eventum/include/class.mime_helper.php	2006-09-08 02:01:57.633010451 +0300
@@ -630,9 +630,9 @@
      * @access  public
      * @param   object $obj The decoded object structure of the MIME message
      * @param   array $parts The parsed parts of the MIME message
-     * @return  array List of parts that exist in the MIME message
+     * @return  void
      */
-    function parse_output(&$obj, &$parts)
+    function parse_output($obj, &$parts)
     {
         if (!empty($obj->parts)) {
             for ($i = 0; $i < count($obj->parts); $i++) {
