--- eventum/misc/irc/bot.php~	2007-01-30 18:59:36.349169654 +0200
+++ eventum/misc/irc/bot.php	2007-01-30 19:00:12.409985378 +0200
@@ -30,13 +30,8 @@
 
 ini_set('memory_limit', '256M');
 
-require_once('../../init.php');
-
-if (!file_exists(APP_CONFIG_PATH . 'irc_config.php')) {
-    echo "ERROR: No config specified. Please see setup/irc_config.php for config information.\n\n";
-    exit;
-}
-
+require_once('/etc/webapps/eventum/core.php');
+require_once('/etc/eventum/irc.php');
 
 // ============================================
 // ============================================
