--- ./include/class.time_tracking.php~	2005-12-30 01:23:58.000000000 +0200
+++ ./include/class.time_tracking.php	2005-12-30 19:20:49.000000000 +0200
@@ -533,7 +533,7 @@
                     ttr_created_date BETWEEN '" . Misc::escapeString($start) . "' AND '" . Misc::escapeString($end) . "'
                  GROUP BY
                     ttc_title";
-        $res = $GLOBALS["db_api"]->dbh->getAssoc($stmt, '', '', DB_FETCHMODE_ASSOC);
+        $res = $GLOBALS["db_api"]->dbh->getAssoc($stmt, false, array(), DB_FETCHMODE_ASSOC);
         if (PEAR::isError($res)) {
             Error_Handler::logError(array($res->getMessage(), $res->getDebugInfo()), __FILE__, __LINE__);
             return array();
