--- ./Date/TimeZone.php~	2005-01-05 18:45:48.000000000 +0200
+++ ./Date/TimeZone.php	2005-04-13 11:44:27.000000000 +0300
@@ -2545,7 +2545,9 @@
         'offset' => 7200000,
         'longname' => "Eastern European Time",
         'shortname' => 'EET',
-        'hasdst' => false ),
+        'hasdst' => true,
+        'dstlongname' => "Eastern European Summer Time",
+        'dstshortname' => 'EEST' ),
     'Europe/Tiraspol' => array(
         'offset' => 7200000,
         'longname' => "Eastern European Time",
@@ -2564,7 +2566,9 @@
         'offset' => 7200000,
         'longname' => "Eastern European Time",
         'shortname' => 'EET',
-        'hasdst' => false ),
+        'hasdst' => true,
+        'dstlongname' => "Eastern European Summer Time",
+        'dstshortname' => 'EEST' ),
     'Europe/Zaporozhye' => array(
         'offset' => 7200000,
         'longname' => "Eastern European Time",
